{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene Expression (SARS, MERS, SARS2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Download RAW data and series matrix from GEO\n",
    "- Microarray \n",
    " * [GSE28166](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE28166), [raw data](https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE28166&format=file), [series matrix](https://ftp.ncbi.nlm.nih.gov/geo/series/GSE28nnn/GSE28166/matrix/GSE28166_series_matrix.txt.gz)\n",
    " * [GSE33267](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE33267), [raw data](https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE33267&format=file), [series matrix](https://ftp.ncbi.nlm.nih.gov/geo/series/GSE33nnn/GSE33267/matrix/GSE33267_series_matrix.txt.gz)\n",
    " * [GSE37571](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE37571), [raw data](https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE37571&format=file), [series matrix](https://ftp.ncbi.nlm.nih.gov/geo/series/GSE37nnn/GSE37571/matrix/GSE37571_series_matrix.txt.gz)\n",
    " * [GSE45042](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE45042), [raw data](https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE45042&format=file), [series matrix](https://ftp.ncbi.nlm.nih.gov/geo/series/GSE45nnn/GSE45042/matrix/GSE45042_series_matrix.txt.gz)\n",
    " * [GSE56677](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE56677), [raw data](https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE56677&format=file), [series matrix](https://ftp.ncbi.nlm.nih.gov/geo/series/GSE56nnn/GSE56677/matrix/GSE56677_series_matrix.txt.gz)\n",
    "- RNASeq\n",
    " * [GSE147507](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE147507) (Human), [raw data](https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE147507&format=file&file=GSE147507%5FRawReadCounts%5FHuman%2Etsv%2Egz), [GSE147507-GPL18573_series_matrix](https://ftp.ncbi.nlm.nih.gov/geo/series/GSE147nnn/GSE147507/matrix/GSE147507-GPL18573_series_matrix.txt.gz)\n",
    " * [GSE148729](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE148729) (totalRNA), [raw data](https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE148729&format=file&file=GSE148729%5FCalu3%5FtotalRNA%5Freadcounts%2Etsv%2Egz), [GSE148729-GPL20301_series_matrix](https://ftp.ncbi.nlm.nih.gov/geo/series/GSE148nnn/GSE148729/matrix/GSE148729-GPL20301_series_matrix.txt.gz)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "from sklearn.decomposition import PCA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from matplotlib import pyplot as plt\n",
    "import seaborn as sns\n",
    "sns.set_style(\"whitegrid\")\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import subprocess as sp"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Microarray"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_properties_to_metadata(data, sep='_', properties={} ):\n",
    "    for prop,position in properties.items():\n",
    "        data[prop] = data['ID'].apply(lambda x: x.split(sep)[position])\n",
    "    if 'Time' in properties:\n",
    "        data['Time']=data['Time'].str.upper().apply(lambda x: x[:-1] if x[-1]=='R' else x)\n",
    "    \n",
    "    return data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_metadata_structure(data):\n",
    "    data=pd.DataFrame(data.loc['!Sample_geo_accession'])\n",
    "    data['ID']=data.index\n",
    "    data.index=data['!Sample_geo_accession']\n",
    "    del data['!Sample_geo_accession']\n",
    "    \n",
    "    return data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def save_metadata_microarray(accession, firstrow, virus='not_mock', sep='_', properties = {'Virus':1, 'Time':2, 'Replicate':3}):\n",
    "    \"\"\"\n",
    "    Save Microarray metadata to csv (accession_meta.csv)\n",
    "    :param accession: geo accession lowercase\n",
    "    :param firstrow: read_csv skip first x row\n",
    "    :param virus: if virus is not mock set virus name (eg. WT)\n",
    "    :param sep: split ID by this separator \n",
    "    :param properties: dictionary: set position of properties (eg. Time) in ID\n",
    "    \"\"\"\n",
    "    data=pd.read_csv(f'../data/{accession.lower()}/{accession.upper()}_series_matrix.txt',\n",
    "                     sep='\\t',skiprows=firstrow,low_memory=False,index_col=0)\n",
    "    data = create_metadata_structure(data)\n",
    "    data = add_properties_to_metadata(data, sep, properties)\n",
    "    \n",
    "    data['Virus'].replace('mock','Mock',inplace=True)\n",
    "    if virus == 'not_mock': \n",
    "        not_mock = data['Virus']!='Mock'\n",
    "    else: \n",
    "        not_mock = data['Virus']==virus\n",
    "    \n",
    "    data.loc[data.index[not_mock],'Virus']=f'{accession.upper()}_Virus'\n",
    "    data.to_csv(f'../data/{accession.lower()}/{accession.lower()}_meta.csv',sep=',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "save_metadata_microarray('gse28166', 37, properties = {'Virus':0, 'Time':1, 'Replicate':2})\n",
    "save_metadata_microarray('gse33267', 35, sep=' ', virus='WT')\n",
    "save_metadata_microarray('gse37571', 31)\n",
    "save_metadata_microarray('gse45042', 39)\n",
    "save_metadata_microarray('gse56677', 35)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Gene expression"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Normalizing raw data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def runR(script, output = True):\n",
    "    result = sp.run(['/usr/local/bin/Rscript', '--verbose', script], capture_output=True)\n",
    "    print(result)\n",
    "    if result.stderr:\n",
    "        print(\"Error: \")\n",
    "        print(result.stderr)\n",
    "    \n",
    "    if output: \n",
    "        print(\"Result: \")\n",
    "        print(result.stdout)\n",
    "        \n",
    "    return"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CompletedProcess(args=['/usr/local/bin/Rscript', '--verbose', '../code/preprocess_microarray.R'], returncode=0, stdout=b'Read ../data/gse28166/GSE28166_RAW//GSM697564.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697565.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697566.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697567.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697568.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697569.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697570.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697571.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697572.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697573.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697574.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697575.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697576.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697577.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697578.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697579.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697580.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697581.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697582.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697583.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697584.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697585.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697586.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697587.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697588.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697589.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697590.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697591.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697592.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697593.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697594.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697595.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697596.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697597.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697598.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697599.txt.gz \\nArray 1 corrected\\nArray 2 corrected\\nArray 3 corrected\\nArray 4 corrected\\nArray 5 corrected\\nArray 6 corrected\\nArray 7 corrected\\nArray 8 corrected\\nArray 9 corrected\\nArray 10 corrected\\nArray 11 corrected\\nArray 12 corrected\\nArray 13 corrected\\nArray 14 corrected\\nArray 15 corrected\\nArray 16 corrected\\nArray 17 corrected\\nArray 18 corrected\\nArray 19 corrected\\nArray 20 corrected\\nArray 21 corrected\\nArray 22 corrected\\nArray 23 corrected\\nArray 24 corrected\\nArray 25 corrected\\nArray 26 corrected\\nArray 27 corrected\\nArray 28 corrected\\nArray 29 corrected\\nArray 30 corrected\\nArray 31 corrected\\nArray 32 corrected\\nArray 33 corrected\\nArray 34 corrected\\nArray 35 corrected\\nArray 36 corrected\\nRead ../data/gse33267/GSE33267_RAW//GSM823177_US93503719_251485059370_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823178_US93503719_251485059419_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823179_US93503719_251485059994_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823180_US93503719_251485060011_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823181_US93503719_251485060018_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823182_US93503719_251485060021_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823183_US93503719_251485060500_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823184_US93503719_251485060013_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823185_US93503719_251485060019_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823186_US93503719_251485059375_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823187_US93503719_251485060015_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823188_US93503719_251485060500_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823189_US93503719_251485060012_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823190_US93503719_251485060020_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823191_US93503719_251485059417_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823192_US93503719_251485059420_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823193_US93503719_251485060014_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823194_US93503719_251485059994_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823195_US93503719_251485059420_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823196_US93503719_251485059995_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823197_US93503719_251485059371_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823198_US93503719_251485060021_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823199_US93503719_251485059376_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823200_US93503719_251485059372_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823201_US93503719_251485059371_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823202_US93503719_251485060015_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823203_US93503719_251485059369_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823204_US93503719_251485059372_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823205_US93503719_251485059995_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823206_US93503719_251485060019_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823207_US93503719_251485059368_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823208_US93503719_251485060014_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823209_US93503719_251485059419_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823210_US93503719_251485060012_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823211_US93503719_251485060014_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823212_US93503719_251485059371_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823213_US93503719_251485060019_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823214_US93503719_251485060022_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823215_US93503719_251485059418_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823216_US93503719_251485059417_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823217_US93503719_251485060015_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823218_US93503719_251485060011_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823219_US93503719_251485060020_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823220_US93503719_251485060014_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823221_US93503719_251485060022_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823222_US93503719_251485059369_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823223_US93503719_251485059371_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823224_US93503719_251485059366_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823225_US93503719_251485060015_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823226_US93503719_251485060011_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823227_US93503719_251485060019_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823228_US93503719_251485060500_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823229_US93503719_251485059367_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823230_US93503719_251485059419_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823231_US93503719_251485060018_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823232_US93503719_251485059373_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823233_US93503719_251485059370_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823234_US93503719_251485059375_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823235_US93503719_251485060013_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823236_US93503719_251485060018_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823237_US93503719_251485059373_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823238_US93503719_251485059419_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823239_US93503719_251485060018_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823240_US93503719_251485059376_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823241_US93503719_251485060021_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823242_US93503719_251485059995_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823243_US93503719_251485059373_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823244_US93503719_251485059420_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823245_US93503719_251485059417_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823246_US93503719_251485059995_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823247_US93503719_251485059372_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823248_US93503719_251485059368_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823249_US93503719_251485060012_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823250_US93503719_251485059366_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823251_US93503719_251485060020_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823252_US93503719_251485059420_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823253_US93503719_251485059418_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823254_US93503719_251485059376_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823255_US93503719_251485059372_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823256_US93503719_251485059373_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823257_US93503719_251485059994_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823258_US93503719_251485059368_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823259_US93503719_251485059366_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823260_US93503719_251485060021_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823261_US93503719_251485060011_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823262_US93503719_251485060013_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823263_US93503719_251485059368_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823264_US93503719_251485059369_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823265_US93503719_251485059367_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823266_US93503719_251485059366_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823267_US93503719_251485060022_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823268_US93503719_251485059418_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823269_US93503719_251485059376_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823270_US93503719_251485059375_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823271_US93503719_251485059367_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823272_US93503719_251485059370_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823273_US93503719_251485059369_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823274_US93503719_251485060013_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823275_US93503719_251485059418_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nArray 1 corrected\\nArray 2 corrected\\nArray 3 corrected\\nArray 4 corrected\\nArray 5 corrected\\nArray 6 corrected\\nArray 7 corrected\\nArray 8 corrected\\nArray 9 corrected\\nArray 10 corrected\\nArray 11 corrected\\nArray 12 corrected\\nArray 13 corrected\\nArray 14 corrected\\nArray 15 corrected\\nArray 16 corrected\\nArray 17 corrected\\nArray 18 corrected\\nArray 19 corrected\\nArray 20 corrected\\nArray 21 corrected\\nArray 22 corrected\\nArray 23 corrected\\nArray 24 corrected\\nArray 25 corrected\\nArray 26 corrected\\nArray 27 corrected\\nArray 28 corrected\\nArray 29 corrected\\nArray 30 corrected\\nArray 31 corrected\\nArray 32 corrected\\nArray 33 corrected\\nArray 34 corrected\\nArray 35 corrected\\nArray 36 corrected\\nArray 37 corrected\\nArray 38 corrected\\nArray 39 corrected\\nArray 40 corrected\\nArray 41 corrected\\nArray 42 corrected\\nArray 43 corrected\\nArray 44 corrected\\nArray 45 corrected\\nArray 46 corrected\\nArray 47 corrected\\nArray 48 corrected\\nArray 49 corrected\\nArray 50 corrected\\nArray 51 corrected\\nArray 52 corrected\\nArray 53 corrected\\nArray 54 corrected\\nArray 55 corrected\\nArray 56 corrected\\nArray 57 corrected\\nArray 58 corrected\\nArray 59 corrected\\nArray 60 corrected\\nArray 61 corrected\\nArray 62 corrected\\nArray 63 corrected\\nArray 64 corrected\\nArray 65 corrected\\nArray 66 corrected\\nArray 67 corrected\\nArray 68 corrected\\nArray 69 corrected\\nArray 70 corrected\\nArray 71 corrected\\nArray 72 corrected\\nArray 73 corrected\\nArray 74 corrected\\nArray 75 corrected\\nArray 76 corrected\\nArray 77 corrected\\nArray 78 corrected\\nArray 79 corrected\\nArray 80 corrected\\nArray 81 corrected\\nArray 82 corrected\\nArray 83 corrected\\nArray 84 corrected\\nArray 85 corrected\\nArray 86 corrected\\nArray 87 corrected\\nArray 88 corrected\\nArray 89 corrected\\nArray 90 corrected\\nArray 91 corrected\\nArray 92 corrected\\nArray 93 corrected\\nArray 94 corrected\\nArray 95 corrected\\nArray 96 corrected\\nArray 97 corrected\\nArray 98 corrected\\nArray 99 corrected\\nRead ../data/gse37571/GSE37571_RAW//GSM921892_US93503719_251485060197_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921893_US93503719_251485060073_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921894_US93503719_251485060033_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921895_US93503719_251485060176_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921896_US93503719_251485060034_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921897_US93503719_251485060195_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921898_US93503719_251485060196_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921899_US93503719_251485060032_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921900_US93503719_251485060034_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921901_US93503719_251485060075_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921902_US93503719_251485060035_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921903_US93503719_251485060076_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921904_US93503719_251485060035_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921905_US93503719_251485060196_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921906_US93503719_251485060025_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921907_US93503719_251485060073_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921908_US93503719_251485060034_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921909_US93503719_251485060033_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921910_US93503719_251485060035_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921911_US93503719_251485060076_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921912_US93503719_251485060073_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921913_US93503719_251485060033_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921914_US93503719_251485060073_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921915_US93503719_251485060074_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921916_US93503719_251485060025_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921917_US93503719_251485060031_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921918_US93503719_251485060196_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921919_US93503719_251485060032_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921920_US93503719_251485060074_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921921_US93503719_251485060072_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921922_US93503719_251485060031_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921923_US93503719_251485060197_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921924_US93503719_251485060196_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921925_US93503719_251485060197_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921926_US93503719_251485060025_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921927_US93503719_251485060075_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921928_US93503719_251485060031_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921929_US93503719_251485060034_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921930_US93503719_251485060195_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921931_US93503719_251485060032_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921932_US93503719_251485060195_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921933_US93503719_251485060176_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921934_US93503719_251485060072_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921935_US93503719_251485060076_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921936_US93503719_251485060074_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921937_US93503719_251485060072_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921938_US93503719_251485060031_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921939_US93503719_251485060075_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921940_US93503719_251485060076_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921941_US93503719_251485060032_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921942_US93503719_251485060195_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921943_US93503719_251485060176_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921944_US93503719_251485060072_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921945_US93503719_251485060074_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nArray 1 corrected\\nArray 2 corrected\\nArray 3 corrected\\nArray 4 corrected\\nArray 5 corrected\\nArray 6 corrected\\nArray 7 corrected\\nArray 8 corrected\\nArray 9 corrected\\nArray 10 corrected\\nArray 11 corrected\\nArray 12 corrected\\nArray 13 corrected\\nArray 14 corrected\\nArray 15 corrected\\nArray 16 corrected\\nArray 17 corrected\\nArray 18 corrected\\nArray 19 corrected\\nArray 20 corrected\\nArray 21 corrected\\nArray 22 corrected\\nArray 23 corrected\\nArray 24 corrected\\nArray 25 corrected\\nArray 26 corrected\\nArray 27 corrected\\nArray 28 corrected\\nArray 29 corrected\\nArray 30 corrected\\nArray 31 corrected\\nArray 32 corrected\\nArray 33 corrected\\nArray 34 corrected\\nArray 35 corrected\\nArray 36 corrected\\nArray 37 corrected\\nArray 38 corrected\\nArray 39 corrected\\nArray 40 corrected\\nArray 41 corrected\\nArray 42 corrected\\nArray 43 corrected\\nArray 44 corrected\\nArray 45 corrected\\nArray 46 corrected\\nArray 47 corrected\\nArray 48 corrected\\nArray 49 corrected\\nArray 50 corrected\\nArray 51 corrected\\nArray 52 corrected\\nArray 53 corrected\\nArray 54 corrected\\nRead ../data/gse45042/GSE45042_RAW//GSM1096511_US93503719_251485073364_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096512_US93503719_251485073364_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096513_US93503719_251485073364_S01_GE1_107_Sep09_1_4.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096514_US93503719_251485073365_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096515_US93503719_251485073365_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096516_US93503719_251485073365_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096517_US93503719_251485073365_S01_GE1_107_Sep09_1_4.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096518_US93503719_251485073366_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096519_US93503719_251485073366_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096520_US93503719_251485073366_S01_GE1_107_Sep09_1_4.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096521_US93503719_251485073367_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096522_US93503719_251485073367_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096523_US93503719_251485073367_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096524_US93503719_251485073367_S01_GE1_107_Sep09_1_4.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096525_US93503719_251485073368_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096526_US93503719_251485073368_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096527_US93503719_251485073368_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096528_US93503719_251485073369_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096529_US93503719_251485073369_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096530_US93503719_251485073369_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096531_US93503719_251485073370_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096532_US93503719_251485073370_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096533_US93503719_251485073370_S01_GE1_107_Sep09_1_4.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096534_US93503719_251485073371_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096535_US93503719_251485073371_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096536_US93503719_251485073371_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096537_US93503719_251485073371_S01_GE1_107_Sep09_1_4.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096538_US93503719_251485073372_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096539_US93503719_251485073372_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096540_US93503719_251485073373_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096541_US93503719_251485073373_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096542_US93503719_251485073373_S01_GE1_107_Sep09_1_3.txt.gz \\nArray 1 corrected\\nArray 2 corrected\\nArray 3 corrected\\nArray 4 corrected\\nArray 5 corrected\\nArray 6 corrected\\nArray 7 corrected\\nArray 8 corrected\\nArray 9 corrected\\nArray 10 corrected\\nArray 11 corrected\\nArray 12 corrected\\nArray 13 corrected\\nArray 14 corrected\\nArray 15 corrected\\nArray 16 corrected\\nArray 17 corrected\\nArray 18 corrected\\nArray 19 corrected\\nArray 20 corrected\\nArray 21 corrected\\nArray 22 corrected\\nArray 23 corrected\\nArray 24 corrected\\nArray 25 corrected\\nArray 26 corrected\\nArray 27 corrected\\nArray 28 corrected\\nArray 29 corrected\\nArray 30 corrected\\nArray 31 corrected\\nArray 32 corrected\\nRead ../data/gse56677/GSE56677_RAW//GSM1366711_US93503719_253949416157_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366712_US93503719_253949416157_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366713_US93503719_253949416157_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366714_US93503719_253949416157_S01_GE1_107_Sep09_2_3.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366715_US93503719_253949416157_S01_GE1_107_Sep09_2_4.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366716_US93503719_253949416160_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366717_US93503719_253949416160_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366718_US93503719_253949416160_S01_GE1_107_Sep09_2_1.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366719_US93503719_253949416160_S01_GE1_107_Sep09_2_2.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366720_US93503719_253949416160_S01_GE1_107_Sep09_2_4.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366721_US93503719_253949416180_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366722_US93503719_253949416180_S01_GE1_107_Sep09_1_4.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366723_US93503719_253949416180_S01_GE1_107_Sep09_2_1.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366724_US93503719_253949416181_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366725_US93503719_253949416181_S01_GE1_107_Sep09_1_4.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366726_US93503719_253949416181_S01_GE1_107_Sep09_2_2.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366727_US93503719_253949416181_S01_GE1_107_Sep09_2_3.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366728_US93503719_253949416181_S01_GE1_107_Sep09_2_4.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366729_US93503719_253949416195_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366730_US93503719_253949416195_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366731_US93503719_253949416195_S01_GE1_107_Sep09_1_4.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366732_US93503719_253949416195_S01_GE1_107_Sep09_2_1.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366733_US93503719_253949416195_S01_GE1_107_Sep09_2_2.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366734_US93503719_253949416195_S01_GE1_107_Sep09_2_3.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366735_US93503719_253949416196_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366736_US93503719_253949416196_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366737_US93503719_253949416196_S01_GE1_107_Sep09_1_4.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366738_US93503719_253949416196_S01_GE1_107_Sep09_2_1.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366739_US93503719_253949416196_S01_GE1_107_Sep09_2_4.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366740_US93503719_253949416197_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366741_US93503719_253949416197_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366742_US93503719_253949416197_S01_GE1_107_Sep09_2_2.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366743_US93503719_253949416197_S01_GE1_107_Sep09_2_3.txt.gz \\nArray 1 corrected\\nArray 2 corrected\\nArray 3 corrected\\nArray 4 corrected\\nArray 5 corrected\\nArray 6 corrected\\nArray 7 corrected\\nArray 8 corrected\\nArray 9 corrected\\nArray 10 corrected\\nArray 11 corrected\\nArray 12 corrected\\nArray 13 corrected\\nArray 14 corrected\\nArray 15 corrected\\nArray 16 corrected\\nArray 17 corrected\\nArray 18 corrected\\nArray 19 corrected\\nArray 20 corrected\\nArray 21 corrected\\nArray 22 corrected\\nArray 23 corrected\\nArray 24 corrected\\nArray 25 corrected\\nArray 26 corrected\\nArray 27 corrected\\nArray 28 corrected\\nArray 29 corrected\\nArray 30 corrected\\nArray 31 corrected\\nArray 32 corrected\\nArray 33 corrected\\n[1] \"SUCCESS\"\\n', stderr=b'running\\n  \\'/Library/Frameworks/R.framework/Versions/4.0/Resources/bin/R --no-echo --no-restore --file=../code/preprocess_microarray.R\\'\\n\\nLoading required package: AnnotationDbi\\nLoading required package: stats4\\nLoading required package: BiocGenerics\\nLoading required package: parallel\\n\\nAttaching package: \\xe2\\x80\\x98BiocGenerics\\xe2\\x80\\x99\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:parallel\\xe2\\x80\\x99:\\n\\n    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\\n    clusterExport, clusterMap, parApply, parCapply, parLapply,\\n    parLapplyLB, parRapply, parSapply, parSapplyLB\\n\\nThe following object is masked from \\xe2\\x80\\x98package:limma\\xe2\\x80\\x99:\\n\\n    plotMA\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:stats\\xe2\\x80\\x99:\\n\\n    IQR, mad, sd, var, xtabs\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:base\\xe2\\x80\\x99:\\n\\n    Filter, Find, Map, Position, Reduce, anyDuplicated, append,\\n    as.data.frame, basename, cbind, colnames, dirname, do.call,\\n    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,\\n    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,\\n    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,\\n    tapply, union, unique, unsplit, which, which.max, which.min\\n\\nLoading required package: Biobase\\nWelcome to Bioconductor\\n\\n    Vignettes contain introductory material; view with\\n    \\'browseVignettes()\\'. To cite Bioconductor, see\\n    \\'citation(\"Biobase\")\\', and for packages \\'citation(\"pkgname\")\\'.\\n\\nLoading required package: IRanges\\nLoading required package: S4Vectors\\n\\nAttaching package: \\xe2\\x80\\x98S4Vectors\\xe2\\x80\\x99\\n\\nThe following object is masked from \\xe2\\x80\\x98package:base\\xe2\\x80\\x99:\\n\\n    expand.grid\\n\\nLoading required package: org.Hs.eg.db\\n\\n\\n\\'select()\\' returned 1:1 mapping between keys and columns\\n\\'select()\\' returned 1:1 mapping between keys and columns\\n\\'select()\\' returned 1:1 mapping between keys and columns\\n\\'select()\\' returned 1:1 mapping between keys and columns\\n\\'select()\\' returned 1:1 mapping between keys and columns\\n')\n",
      "Error: \n",
      "b'running\\n  \\'/Library/Frameworks/R.framework/Versions/4.0/Resources/bin/R --no-echo --no-restore --file=../code/preprocess_microarray.R\\'\\n\\nLoading required package: AnnotationDbi\\nLoading required package: stats4\\nLoading required package: BiocGenerics\\nLoading required package: parallel\\n\\nAttaching package: \\xe2\\x80\\x98BiocGenerics\\xe2\\x80\\x99\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:parallel\\xe2\\x80\\x99:\\n\\n    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\\n    clusterExport, clusterMap, parApply, parCapply, parLapply,\\n    parLapplyLB, parRapply, parSapply, parSapplyLB\\n\\nThe following object is masked from \\xe2\\x80\\x98package:limma\\xe2\\x80\\x99:\\n\\n    plotMA\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:stats\\xe2\\x80\\x99:\\n\\n    IQR, mad, sd, var, xtabs\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:base\\xe2\\x80\\x99:\\n\\n    Filter, Find, Map, Position, Reduce, anyDuplicated, append,\\n    as.data.frame, basename, cbind, colnames, dirname, do.call,\\n    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,\\n    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,\\n    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,\\n    tapply, union, unique, unsplit, which, which.max, which.min\\n\\nLoading required package: Biobase\\nWelcome to Bioconductor\\n\\n    Vignettes contain introductory material; view with\\n    \\'browseVignettes()\\'. To cite Bioconductor, see\\n    \\'citation(\"Biobase\")\\', and for packages \\'citation(\"pkgname\")\\'.\\n\\nLoading required package: IRanges\\nLoading required package: S4Vectors\\n\\nAttaching package: \\xe2\\x80\\x98S4Vectors\\xe2\\x80\\x99\\n\\nThe following object is masked from \\xe2\\x80\\x98package:base\\xe2\\x80\\x99:\\n\\n    expand.grid\\n\\nLoading required package: org.Hs.eg.db\\n\\n\\n\\'select()\\' returned 1:1 mapping between keys and columns\\n\\'select()\\' returned 1:1 mapping between keys and columns\\n\\'select()\\' returned 1:1 mapping between keys and columns\\n\\'select()\\' returned 1:1 mapping between keys and columns\\n\\'select()\\' returned 1:1 mapping between keys and columns\\n'\n",
      "Result: \n",
      "b'Read ../data/gse28166/GSE28166_RAW//GSM697564.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697565.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697566.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697567.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697568.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697569.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697570.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697571.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697572.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697573.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697574.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697575.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697576.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697577.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697578.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697579.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697580.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697581.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697582.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697583.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697584.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697585.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697586.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697587.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697588.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697589.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697590.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697591.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697592.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697593.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697594.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697595.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697596.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697597.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697598.txt.gz \\nRead ../data/gse28166/GSE28166_RAW//GSM697599.txt.gz \\nArray 1 corrected\\nArray 2 corrected\\nArray 3 corrected\\nArray 4 corrected\\nArray 5 corrected\\nArray 6 corrected\\nArray 7 corrected\\nArray 8 corrected\\nArray 9 corrected\\nArray 10 corrected\\nArray 11 corrected\\nArray 12 corrected\\nArray 13 corrected\\nArray 14 corrected\\nArray 15 corrected\\nArray 16 corrected\\nArray 17 corrected\\nArray 18 corrected\\nArray 19 corrected\\nArray 20 corrected\\nArray 21 corrected\\nArray 22 corrected\\nArray 23 corrected\\nArray 24 corrected\\nArray 25 corrected\\nArray 26 corrected\\nArray 27 corrected\\nArray 28 corrected\\nArray 29 corrected\\nArray 30 corrected\\nArray 31 corrected\\nArray 32 corrected\\nArray 33 corrected\\nArray 34 corrected\\nArray 35 corrected\\nArray 36 corrected\\nRead ../data/gse33267/GSE33267_RAW//GSM823177_US93503719_251485059370_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823178_US93503719_251485059419_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823179_US93503719_251485059994_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823180_US93503719_251485060011_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823181_US93503719_251485060018_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823182_US93503719_251485060021_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823183_US93503719_251485060500_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823184_US93503719_251485060013_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823185_US93503719_251485060019_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823186_US93503719_251485059375_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823187_US93503719_251485060015_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823188_US93503719_251485060500_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823189_US93503719_251485060012_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823190_US93503719_251485060020_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823191_US93503719_251485059417_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823192_US93503719_251485059420_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823193_US93503719_251485060014_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823194_US93503719_251485059994_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823195_US93503719_251485059420_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823196_US93503719_251485059995_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823197_US93503719_251485059371_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823198_US93503719_251485060021_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823199_US93503719_251485059376_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823200_US93503719_251485059372_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823201_US93503719_251485059371_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823202_US93503719_251485060015_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823203_US93503719_251485059369_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823204_US93503719_251485059372_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823205_US93503719_251485059995_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823206_US93503719_251485060019_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823207_US93503719_251485059368_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823208_US93503719_251485060014_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823209_US93503719_251485059419_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823210_US93503719_251485060012_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823211_US93503719_251485060014_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823212_US93503719_251485059371_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823213_US93503719_251485060019_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823214_US93503719_251485060022_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823215_US93503719_251485059418_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823216_US93503719_251485059417_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823217_US93503719_251485060015_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823218_US93503719_251485060011_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823219_US93503719_251485060020_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823220_US93503719_251485060014_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823221_US93503719_251485060022_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823222_US93503719_251485059369_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823223_US93503719_251485059371_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823224_US93503719_251485059366_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823225_US93503719_251485060015_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823226_US93503719_251485060011_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823227_US93503719_251485060019_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823228_US93503719_251485060500_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823229_US93503719_251485059367_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823230_US93503719_251485059419_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823231_US93503719_251485060018_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823232_US93503719_251485059373_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823233_US93503719_251485059370_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823234_US93503719_251485059375_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823235_US93503719_251485060013_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823236_US93503719_251485060018_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823237_US93503719_251485059373_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823238_US93503719_251485059419_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823239_US93503719_251485060018_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823240_US93503719_251485059376_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823241_US93503719_251485060021_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823242_US93503719_251485059995_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823243_US93503719_251485059373_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823244_US93503719_251485059420_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823245_US93503719_251485059417_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823246_US93503719_251485059995_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823247_US93503719_251485059372_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823248_US93503719_251485059368_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823249_US93503719_251485060012_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823250_US93503719_251485059366_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823251_US93503719_251485060020_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823252_US93503719_251485059420_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823253_US93503719_251485059418_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823254_US93503719_251485059376_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823255_US93503719_251485059372_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823256_US93503719_251485059373_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823257_US93503719_251485059994_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823258_US93503719_251485059368_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823259_US93503719_251485059366_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823260_US93503719_251485060021_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823261_US93503719_251485060011_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823262_US93503719_251485060013_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823263_US93503719_251485059368_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823264_US93503719_251485059369_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823265_US93503719_251485059367_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823266_US93503719_251485059366_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823267_US93503719_251485060022_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823268_US93503719_251485059418_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823269_US93503719_251485059376_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823270_US93503719_251485059375_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823271_US93503719_251485059367_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823272_US93503719_251485059370_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823273_US93503719_251485059369_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823274_US93503719_251485060013_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse33267/GSE33267_RAW//GSM823275_US93503719_251485059418_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nArray 1 corrected\\nArray 2 corrected\\nArray 3 corrected\\nArray 4 corrected\\nArray 5 corrected\\nArray 6 corrected\\nArray 7 corrected\\nArray 8 corrected\\nArray 9 corrected\\nArray 10 corrected\\nArray 11 corrected\\nArray 12 corrected\\nArray 13 corrected\\nArray 14 corrected\\nArray 15 corrected\\nArray 16 corrected\\nArray 17 corrected\\nArray 18 corrected\\nArray 19 corrected\\nArray 20 corrected\\nArray 21 corrected\\nArray 22 corrected\\nArray 23 corrected\\nArray 24 corrected\\nArray 25 corrected\\nArray 26 corrected\\nArray 27 corrected\\nArray 28 corrected\\nArray 29 corrected\\nArray 30 corrected\\nArray 31 corrected\\nArray 32 corrected\\nArray 33 corrected\\nArray 34 corrected\\nArray 35 corrected\\nArray 36 corrected\\nArray 37 corrected\\nArray 38 corrected\\nArray 39 corrected\\nArray 40 corrected\\nArray 41 corrected\\nArray 42 corrected\\nArray 43 corrected\\nArray 44 corrected\\nArray 45 corrected\\nArray 46 corrected\\nArray 47 corrected\\nArray 48 corrected\\nArray 49 corrected\\nArray 50 corrected\\nArray 51 corrected\\nArray 52 corrected\\nArray 53 corrected\\nArray 54 corrected\\nArray 55 corrected\\nArray 56 corrected\\nArray 57 corrected\\nArray 58 corrected\\nArray 59 corrected\\nArray 60 corrected\\nArray 61 corrected\\nArray 62 corrected\\nArray 63 corrected\\nArray 64 corrected\\nArray 65 corrected\\nArray 66 corrected\\nArray 67 corrected\\nArray 68 corrected\\nArray 69 corrected\\nArray 70 corrected\\nArray 71 corrected\\nArray 72 corrected\\nArray 73 corrected\\nArray 74 corrected\\nArray 75 corrected\\nArray 76 corrected\\nArray 77 corrected\\nArray 78 corrected\\nArray 79 corrected\\nArray 80 corrected\\nArray 81 corrected\\nArray 82 corrected\\nArray 83 corrected\\nArray 84 corrected\\nArray 85 corrected\\nArray 86 corrected\\nArray 87 corrected\\nArray 88 corrected\\nArray 89 corrected\\nArray 90 corrected\\nArray 91 corrected\\nArray 92 corrected\\nArray 93 corrected\\nArray 94 corrected\\nArray 95 corrected\\nArray 96 corrected\\nArray 97 corrected\\nArray 98 corrected\\nArray 99 corrected\\nRead ../data/gse37571/GSE37571_RAW//GSM921892_US93503719_251485060197_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921893_US93503719_251485060073_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921894_US93503719_251485060033_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921895_US93503719_251485060176_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921896_US93503719_251485060034_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921897_US93503719_251485060195_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921898_US93503719_251485060196_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921899_US93503719_251485060032_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921900_US93503719_251485060034_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921901_US93503719_251485060075_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921902_US93503719_251485060035_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921903_US93503719_251485060076_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921904_US93503719_251485060035_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921905_US93503719_251485060196_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921906_US93503719_251485060025_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921907_US93503719_251485060073_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921908_US93503719_251485060034_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921909_US93503719_251485060033_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921910_US93503719_251485060035_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921911_US93503719_251485060076_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921912_US93503719_251485060073_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921913_US93503719_251485060033_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921914_US93503719_251485060073_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921915_US93503719_251485060074_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921916_US93503719_251485060025_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921917_US93503719_251485060031_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921918_US93503719_251485060196_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921919_US93503719_251485060032_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921920_US93503719_251485060074_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921921_US93503719_251485060072_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921922_US93503719_251485060031_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921923_US93503719_251485060197_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921924_US93503719_251485060196_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921925_US93503719_251485060197_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921926_US93503719_251485060025_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921927_US93503719_251485060075_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921928_US93503719_251485060031_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921929_US93503719_251485060034_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921930_US93503719_251485060195_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921931_US93503719_251485060032_S01_GE1-v5_95_Feb07_1_1.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921932_US93503719_251485060195_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921933_US93503719_251485060176_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921934_US93503719_251485060072_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921935_US93503719_251485060076_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921936_US93503719_251485060074_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921937_US93503719_251485060072_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921938_US93503719_251485060031_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921939_US93503719_251485060075_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921940_US93503719_251485060076_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921941_US93503719_251485060032_S01_GE1-v5_95_Feb07_1_4.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921942_US93503719_251485060195_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921943_US93503719_251485060176_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921944_US93503719_251485060072_S01_GE1-v5_95_Feb07_1_3.txt.gz \\nRead ../data/gse37571/GSE37571_RAW//GSM921945_US93503719_251485060074_S01_GE1-v5_95_Feb07_1_2.txt.gz \\nArray 1 corrected\\nArray 2 corrected\\nArray 3 corrected\\nArray 4 corrected\\nArray 5 corrected\\nArray 6 corrected\\nArray 7 corrected\\nArray 8 corrected\\nArray 9 corrected\\nArray 10 corrected\\nArray 11 corrected\\nArray 12 corrected\\nArray 13 corrected\\nArray 14 corrected\\nArray 15 corrected\\nArray 16 corrected\\nArray 17 corrected\\nArray 18 corrected\\nArray 19 corrected\\nArray 20 corrected\\nArray 21 corrected\\nArray 22 corrected\\nArray 23 corrected\\nArray 24 corrected\\nArray 25 corrected\\nArray 26 corrected\\nArray 27 corrected\\nArray 28 corrected\\nArray 29 corrected\\nArray 30 corrected\\nArray 31 corrected\\nArray 32 corrected\\nArray 33 corrected\\nArray 34 corrected\\nArray 35 corrected\\nArray 36 corrected\\nArray 37 corrected\\nArray 38 corrected\\nArray 39 corrected\\nArray 40 corrected\\nArray 41 corrected\\nArray 42 corrected\\nArray 43 corrected\\nArray 44 corrected\\nArray 45 corrected\\nArray 46 corrected\\nArray 47 corrected\\nArray 48 corrected\\nArray 49 corrected\\nArray 50 corrected\\nArray 51 corrected\\nArray 52 corrected\\nArray 53 corrected\\nArray 54 corrected\\nRead ../data/gse45042/GSE45042_RAW//GSM1096511_US93503719_251485073364_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096512_US93503719_251485073364_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096513_US93503719_251485073364_S01_GE1_107_Sep09_1_4.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096514_US93503719_251485073365_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096515_US93503719_251485073365_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096516_US93503719_251485073365_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096517_US93503719_251485073365_S01_GE1_107_Sep09_1_4.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096518_US93503719_251485073366_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096519_US93503719_251485073366_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096520_US93503719_251485073366_S01_GE1_107_Sep09_1_4.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096521_US93503719_251485073367_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096522_US93503719_251485073367_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096523_US93503719_251485073367_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096524_US93503719_251485073367_S01_GE1_107_Sep09_1_4.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096525_US93503719_251485073368_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096526_US93503719_251485073368_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096527_US93503719_251485073368_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096528_US93503719_251485073369_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096529_US93503719_251485073369_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096530_US93503719_251485073369_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096531_US93503719_251485073370_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096532_US93503719_251485073370_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096533_US93503719_251485073370_S01_GE1_107_Sep09_1_4.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096534_US93503719_251485073371_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096535_US93503719_251485073371_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096536_US93503719_251485073371_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096537_US93503719_251485073371_S01_GE1_107_Sep09_1_4.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096538_US93503719_251485073372_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096539_US93503719_251485073372_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096540_US93503719_251485073373_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096541_US93503719_251485073373_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse45042/GSE45042_RAW//GSM1096542_US93503719_251485073373_S01_GE1_107_Sep09_1_3.txt.gz \\nArray 1 corrected\\nArray 2 corrected\\nArray 3 corrected\\nArray 4 corrected\\nArray 5 corrected\\nArray 6 corrected\\nArray 7 corrected\\nArray 8 corrected\\nArray 9 corrected\\nArray 10 corrected\\nArray 11 corrected\\nArray 12 corrected\\nArray 13 corrected\\nArray 14 corrected\\nArray 15 corrected\\nArray 16 corrected\\nArray 17 corrected\\nArray 18 corrected\\nArray 19 corrected\\nArray 20 corrected\\nArray 21 corrected\\nArray 22 corrected\\nArray 23 corrected\\nArray 24 corrected\\nArray 25 corrected\\nArray 26 corrected\\nArray 27 corrected\\nArray 28 corrected\\nArray 29 corrected\\nArray 30 corrected\\nArray 31 corrected\\nArray 32 corrected\\nRead ../data/gse56677/GSE56677_RAW//GSM1366711_US93503719_253949416157_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366712_US93503719_253949416157_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366713_US93503719_253949416157_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366714_US93503719_253949416157_S01_GE1_107_Sep09_2_3.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366715_US93503719_253949416157_S01_GE1_107_Sep09_2_4.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366716_US93503719_253949416160_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366717_US93503719_253949416160_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366718_US93503719_253949416160_S01_GE1_107_Sep09_2_1.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366719_US93503719_253949416160_S01_GE1_107_Sep09_2_2.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366720_US93503719_253949416160_S01_GE1_107_Sep09_2_4.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366721_US93503719_253949416180_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366722_US93503719_253949416180_S01_GE1_107_Sep09_1_4.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366723_US93503719_253949416180_S01_GE1_107_Sep09_2_1.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366724_US93503719_253949416181_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366725_US93503719_253949416181_S01_GE1_107_Sep09_1_4.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366726_US93503719_253949416181_S01_GE1_107_Sep09_2_2.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366727_US93503719_253949416181_S01_GE1_107_Sep09_2_3.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366728_US93503719_253949416181_S01_GE1_107_Sep09_2_4.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366729_US93503719_253949416195_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366730_US93503719_253949416195_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366731_US93503719_253949416195_S01_GE1_107_Sep09_1_4.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366732_US93503719_253949416195_S01_GE1_107_Sep09_2_1.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366733_US93503719_253949416195_S01_GE1_107_Sep09_2_2.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366734_US93503719_253949416195_S01_GE1_107_Sep09_2_3.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366735_US93503719_253949416196_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366736_US93503719_253949416196_S01_GE1_107_Sep09_1_2.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366737_US93503719_253949416196_S01_GE1_107_Sep09_1_4.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366738_US93503719_253949416196_S01_GE1_107_Sep09_2_1.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366739_US93503719_253949416196_S01_GE1_107_Sep09_2_4.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366740_US93503719_253949416197_S01_GE1_107_Sep09_1_1.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366741_US93503719_253949416197_S01_GE1_107_Sep09_1_3.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366742_US93503719_253949416197_S01_GE1_107_Sep09_2_2.txt.gz \\nRead ../data/gse56677/GSE56677_RAW//GSM1366743_US93503719_253949416197_S01_GE1_107_Sep09_2_3.txt.gz \\nArray 1 corrected\\nArray 2 corrected\\nArray 3 corrected\\nArray 4 corrected\\nArray 5 corrected\\nArray 6 corrected\\nArray 7 corrected\\nArray 8 corrected\\nArray 9 corrected\\nArray 10 corrected\\nArray 11 corrected\\nArray 12 corrected\\nArray 13 corrected\\nArray 14 corrected\\nArray 15 corrected\\nArray 16 corrected\\nArray 17 corrected\\nArray 18 corrected\\nArray 19 corrected\\nArray 20 corrected\\nArray 21 corrected\\nArray 22 corrected\\nArray 23 corrected\\nArray 24 corrected\\nArray 25 corrected\\nArray 26 corrected\\nArray 27 corrected\\nArray 28 corrected\\nArray 29 corrected\\nArray 30 corrected\\nArray 31 corrected\\nArray 32 corrected\\nArray 33 corrected\\n[1] \"SUCCESS\"\\n'\n"
     ]
    }
   ],
   "source": [
    "runR(\"../code/preprocess_microarray.R\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Save normalized data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "def save_gene_expression_microarray(accession, split_by='_'):\n",
    "    \"\"\"\n",
    "    Save normalized microarray gene expression data to csv (accession_gex.csv)\n",
    "    :param accession: GEO accession (lowercase)\n",
    "    :param split_by: split data columns by this character\n",
    "    \"\"\"\n",
    "    accession = accession.lower()\n",
    "    data=pd.read_csv(f'../data/{accession}/{accession}_norm_mean.csv',sep=',',header=0,index_col=0)\n",
    "    meta=pd.read_csv(f'../data/{accession}/{accession}_meta.csv',sep=',',header=0,index_col=0)\n",
    "    data.columns=pd.Series(data.columns).apply(lambda x:x.split(split_by)[0])\n",
    "    #assert len(set(data.columns)-set(meta.index))==0\n",
    "    data=data[meta.index]\n",
    "    data.to_csv(f'../data/{accession}/{accession}_gex.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "save_gene_expression_microarray('gse28166', split_by='.')\n",
    "save_gene_expression_microarray('gse33267')\n",
    "save_gene_expression_microarray('gse37571')\n",
    "save_gene_expression_microarray('gse45042')\n",
    "save_gene_expression_microarray('gse56677')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Differential expression analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# runR(\"../code/microarray_DE.R\") - missing?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## RNA Seq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### GSE147507"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [],
   "source": [
    "series_matrix=pd.read_csv('../data/gse147507/GSE147507-GPL18573_series_matrix.txt'\n",
    "                          ,sep='\\t',header=0,index_col=0,skiprows=33)\n",
    "metadata = create_metadata_structure(series_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata = add_properties_to_metadata(metadata, properties = {'Series':0, 'Cell':1, 'Treatment':2})\n",
    "metadata.loc[['GSM4462413', 'GSM4462414', 'GSM4462415', 'GSM4462416'], ['Cell', 'Treatment']] = [['Lung','Healthy'],['Lung','Healthy'],['Lung','COVID19'],['Lung','COVID19']]\n",
    "metadata['Virus']=1\n",
    "metadata.loc[metadata['Treatment'].str.contains('Mock|Healthy', regex=True), 'Virus'] = 0\n",
    "metadata.to_csv('../data/gse147507/meta.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Gene expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "data=pd.read_csv('../data/gse147507/GSE147507_RawReadCounts_Human.tsv',sep='\\t',header=0,index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "data=data[metadata['ID'].values]\n",
    "data.columns=metadata.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "data.to_csv('../data/gse147507/GSE147507_raw.csv',sep=',')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Differential expression analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CompletedProcess(args=['/usr/local/bin/Rscript', '--verbose', '../code/gse147507_DE.R'], returncode=1, stdout=b'', stderr=b'running\\n  \\'/Library/Frameworks/R.framework/Versions/4.0/Resources/bin/R --no-echo --no-restore --file=../code/gse147507_DE.R\\'\\n\\nLoading required package: S4Vectors\\nLoading required package: stats4\\nLoading required package: BiocGenerics\\nLoading required package: parallel\\n\\nAttaching package: \\xe2\\x80\\x98BiocGenerics\\xe2\\x80\\x99\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:parallel\\xe2\\x80\\x99:\\n\\n    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\\n    clusterExport, clusterMap, parApply, parCapply, parLapply,\\n    parLapplyLB, parRapply, parSapply, parSapplyLB\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:stats\\xe2\\x80\\x99:\\n\\n    IQR, mad, sd, var, xtabs\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:base\\xe2\\x80\\x99:\\n\\n    Filter, Find, Map, Position, Reduce, anyDuplicated, append,\\n    as.data.frame, basename, cbind, colnames, dirname, do.call,\\n    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,\\n    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,\\n    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,\\n    tapply, union, unique, unsplit, which, which.max, which.min\\n\\n\\nAttaching package: \\xe2\\x80\\x98S4Vectors\\xe2\\x80\\x99\\n\\nThe following object is masked from \\xe2\\x80\\x98package:base\\xe2\\x80\\x99:\\n\\n    expand.grid\\n\\nLoading required package: IRanges\\nLoading required package: GenomicRanges\\nLoading required package: GenomeInfoDb\\nLoading required package: SummarizedExperiment\\nLoading required package: Biobase\\nWelcome to Bioconductor\\n\\n    Vignettes contain introductory material; view with\\n    \\'browseVignettes()\\'. To cite Bioconductor, see\\n    \\'citation(\"Biobase\")\\', and for packages \\'citation(\"pkgname\")\\'.\\n\\nLoading required package: DelayedArray\\nLoading required package: matrixStats\\n\\nAttaching package: \\xe2\\x80\\x98matrixStats\\xe2\\x80\\x99\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:Biobase\\xe2\\x80\\x99:\\n\\n    anyMissing, rowMedians\\n\\n\\nAttaching package: \\xe2\\x80\\x98DelayedArray\\xe2\\x80\\x99\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:matrixStats\\xe2\\x80\\x99:\\n\\n    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:base\\xe2\\x80\\x99:\\n\\n    aperm, apply, rowsum\\n\\n\\xe2\\x94\\x80\\xe2\\x94\\x80 Attaching packages \\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80 tidyverse 1.3.0 \\xe2\\x94\\x80\\xe2\\x94\\x80\\n\\xe2\\x9c\\x94 ggplot2 3.3.2     \\xe2\\x9c\\x94 purrr   0.3.4\\n\\xe2\\x9c\\x94 tibble  3.0.4     \\xe2\\x9c\\x94 dplyr   1.0.2\\n\\xe2\\x9c\\x94 tidyr   1.1.2     \\xe2\\x9c\\x94 stringr 1.4.0\\n\\xe2\\x9c\\x94 readr   1.4.0     \\xe2\\x9c\\x94 forcats 0.5.0\\n\\xe2\\x94\\x80\\xe2\\x94\\x80 Conflicts \\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80 tidyverse_conflicts() \\xe2\\x94\\x80\\xe2\\x94\\x80\\n\\xe2\\x9c\\x96 ggplot2::Position() masks BiocGenerics::Position(), base::Position()\\n\\xe2\\x9c\\x96 dplyr::collapse()   masks IRanges::collapse()\\n\\xe2\\x9c\\x96 dplyr::combine()    masks Biobase::combine(), BiocGenerics::combine()\\n\\xe2\\x9c\\x96 dplyr::count()      masks matrixStats::count()\\n\\xe2\\x9c\\x96 dplyr::desc()       masks IRanges::desc()\\n\\xe2\\x9c\\x96 tidyr::expand()     masks S4Vectors::expand()\\n\\xe2\\x9c\\x96 dplyr::filter()     masks stats::filter()\\n\\xe2\\x9c\\x96 dplyr::first()      masks S4Vectors::first()\\n\\xe2\\x9c\\x96 dplyr::lag()        masks stats::lag()\\n\\xe2\\x9c\\x96 purrr::reduce()     masks GenomicRanges::reduce(), IRanges::reduce()\\n\\xe2\\x9c\\x96 dplyr::rename()     masks S4Vectors::rename()\\n\\xe2\\x9c\\x96 purrr::simplify()   masks DelayedArray::simplify()\\n\\xe2\\x9c\\x96 dplyr::slice()      masks IRanges::slice()\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\nestimating size factors\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\nestimating dispersions\\ngene-wise dispersion estimates: 10 workers\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\nmean-dispersion relationship\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\nfinal dispersion estimates, fitting model and testing: 10 workers\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n-- replacing outliers and refitting for 763 genes\\n-- DESeq argument \\'minReplicatesForReplace\\' = 7 \\n-- original counts are preserved in counts(dds)\\nestimating dispersions\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\nfitting model and testing\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\nError in file(file, ifelse(append, \"a\", \"w\")) : \\n  cannot open the connection\\nCalls: write.csv -> eval.parent -> eval -> eval -> write.table -> file\\nIn addition: Warning message:\\nIn file(file, ifelse(append, \"a\", \"w\")) :\\n  cannot open file \\'../results/gse147507/gse147507_vst.csv\\': No such file or directory\\nExecution halted\\n')\n",
      "Error: \n",
      "b'running\\n  \\'/Library/Frameworks/R.framework/Versions/4.0/Resources/bin/R --no-echo --no-restore --file=../code/gse147507_DE.R\\'\\n\\nLoading required package: S4Vectors\\nLoading required package: stats4\\nLoading required package: BiocGenerics\\nLoading required package: parallel\\n\\nAttaching package: \\xe2\\x80\\x98BiocGenerics\\xe2\\x80\\x99\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:parallel\\xe2\\x80\\x99:\\n\\n    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\\n    clusterExport, clusterMap, parApply, parCapply, parLapply,\\n    parLapplyLB, parRapply, parSapply, parSapplyLB\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:stats\\xe2\\x80\\x99:\\n\\n    IQR, mad, sd, var, xtabs\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:base\\xe2\\x80\\x99:\\n\\n    Filter, Find, Map, Position, Reduce, anyDuplicated, append,\\n    as.data.frame, basename, cbind, colnames, dirname, do.call,\\n    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,\\n    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,\\n    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,\\n    tapply, union, unique, unsplit, which, which.max, which.min\\n\\n\\nAttaching package: \\xe2\\x80\\x98S4Vectors\\xe2\\x80\\x99\\n\\nThe following object is masked from \\xe2\\x80\\x98package:base\\xe2\\x80\\x99:\\n\\n    expand.grid\\n\\nLoading required package: IRanges\\nLoading required package: GenomicRanges\\nLoading required package: GenomeInfoDb\\nLoading required package: SummarizedExperiment\\nLoading required package: Biobase\\nWelcome to Bioconductor\\n\\n    Vignettes contain introductory material; view with\\n    \\'browseVignettes()\\'. To cite Bioconductor, see\\n    \\'citation(\"Biobase\")\\', and for packages \\'citation(\"pkgname\")\\'.\\n\\nLoading required package: DelayedArray\\nLoading required package: matrixStats\\n\\nAttaching package: \\xe2\\x80\\x98matrixStats\\xe2\\x80\\x99\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:Biobase\\xe2\\x80\\x99:\\n\\n    anyMissing, rowMedians\\n\\n\\nAttaching package: \\xe2\\x80\\x98DelayedArray\\xe2\\x80\\x99\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:matrixStats\\xe2\\x80\\x99:\\n\\n    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:base\\xe2\\x80\\x99:\\n\\n    aperm, apply, rowsum\\n\\n\\xe2\\x94\\x80\\xe2\\x94\\x80 Attaching packages \\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80 tidyverse 1.3.0 \\xe2\\x94\\x80\\xe2\\x94\\x80\\n\\xe2\\x9c\\x94 ggplot2 3.3.2     \\xe2\\x9c\\x94 purrr   0.3.4\\n\\xe2\\x9c\\x94 tibble  3.0.4     \\xe2\\x9c\\x94 dplyr   1.0.2\\n\\xe2\\x9c\\x94 tidyr   1.1.2     \\xe2\\x9c\\x94 stringr 1.4.0\\n\\xe2\\x9c\\x94 readr   1.4.0     \\xe2\\x9c\\x94 forcats 0.5.0\\n\\xe2\\x94\\x80\\xe2\\x94\\x80 Conflicts \\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80 tidyverse_conflicts() \\xe2\\x94\\x80\\xe2\\x94\\x80\\n\\xe2\\x9c\\x96 ggplot2::Position() masks BiocGenerics::Position(), base::Position()\\n\\xe2\\x9c\\x96 dplyr::collapse()   masks IRanges::collapse()\\n\\xe2\\x9c\\x96 dplyr::combine()    masks Biobase::combine(), BiocGenerics::combine()\\n\\xe2\\x9c\\x96 dplyr::count()      masks matrixStats::count()\\n\\xe2\\x9c\\x96 dplyr::desc()       masks IRanges::desc()\\n\\xe2\\x9c\\x96 tidyr::expand()     masks S4Vectors::expand()\\n\\xe2\\x9c\\x96 dplyr::filter()     masks stats::filter()\\n\\xe2\\x9c\\x96 dplyr::first()      masks S4Vectors::first()\\n\\xe2\\x9c\\x96 dplyr::lag()        masks stats::lag()\\n\\xe2\\x9c\\x96 purrr::reduce()     masks GenomicRanges::reduce(), IRanges::reduce()\\n\\xe2\\x9c\\x96 dplyr::rename()     masks S4Vectors::rename()\\n\\xe2\\x9c\\x96 purrr::simplify()   masks DelayedArray::simplify()\\n\\xe2\\x9c\\x96 dplyr::slice()      masks IRanges::slice()\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\nestimating size factors\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\nestimating dispersions\\ngene-wise dispersion estimates: 10 workers\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\nmean-dispersion relationship\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\nfinal dispersion estimates, fitting model and testing: 10 workers\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\n-- replacing outliers and refitting for 763 genes\\n-- DESeq argument \\'minReplicatesForReplace\\' = 7 \\n-- original counts are preserved in counts(dds)\\nestimating dispersions\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\nfitting model and testing\\n  Note: levels of factors in the design contain characters other than\\n  letters, numbers, \\'_\\' and \\'.\\'. It is recommended (but not required) to use\\n  only letters, numbers, and delimiters \\'_\\' or \\'.\\', as these are safe characters\\n  for column names in R. [This is a message, not a warning or an error]\\nError in file(file, ifelse(append, \"a\", \"w\")) : \\n  cannot open the connection\\nCalls: write.csv -> eval.parent -> eval -> eval -> write.table -> file\\nIn addition: Warning message:\\nIn file(file, ifelse(append, \"a\", \"w\")) :\\n  cannot open file \\'../results/gse147507/gse147507_vst.csv\\': No such file or directory\\nExecution halted\\n'\n",
      "Result: \n",
      "b''\n"
     ]
    }
   ],
   "source": [
    "runR(\"../code/gse147507_DE.R\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### PC anaysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "model=PCA(2)\n",
    "data_pca=pd.DataFrame(model.fit_transform(data.T),index=data.columns,columns=['PC1','PC2'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAZ8AAAEECAYAAAD3QzkUAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjMsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+AADFEAAAgAElEQVR4nOzdeXxcxZno/V/vm7q1y7Zs2TJeygID3sALiwFjHDAEEhiGkBhCyMxkXmYSSPKZZObCDbzhhiEJmTd5J2QhhDAkGchKWEIwMAEMNgYLjBfkMl7wblmStfW+nftHS23JWq1udbfk5/v5JKirq8+pU5L1qOrUecpkGAZCCCFELpnz3QAhhBCnHwk+Qgghck6CjxBCiJyT4COEECLnJPgIIYTIOWu+G1AoNm/ebDgcjnw3o1+RSIRCbVshkP4ZmPTN4KR/Bjec/gkGg80LFy6sPNVjS/Dp4nA4qKury3cz+tXQ0FCwbSsE0j8Dk74ZnPTP4IbTP/X19ftGcmyZdhNCCJFzEnyEEELknAQfIYQQOSf3fAYRi8U4ePAg4XA47+1oaGgY0WedTidTpkzBZrNluVVCCDFyEnwGcfDgQbxeL7W1tZhMpry1IxQK4XK5TvlzhmHQ0tLCwYMHmT59+ii0TAghRkam3QYRDocpLy/Pa+DJhMlkory8PO8jNyFOZ/6OAIcPHGXT+s00HmkiHJR/jyAjnyGN1cDTbay3X4ixLBQI89zvXuKHD/4cAIvFzAMP3835F83HYrHkuXX5JSMfIYQYJX5/gJ9877/SrxOJJN++5z9pO96Rx1YVBhn5FLhkMsn999/Prl27sNvt3H///UybNi39/m9+8xuefPJJrFYr//iP/8ill16ax9YKMTpCgRCd7QH26f1MPqMaX6mXIp8n380aUiwWJx6L9yo73tyG7KMmwSerIq0thI4eIhmLYrbZcU2cjKO0PKNjvvzyy0QiEZ566ik2b97Mv//7v/OjH/0IgKamJp544gl+//vfE4lEuPnmm7nggguw2+3ZuBwhCkIinuD9Ddv44d0/S//SvvEfr+OKGy/D6XbmuXWDczod1M6s4aNdB9JlF12+GKdTUvrItFuWRFpbCBzcRzIWBSAZixI4uI9Ia0tGx62vr+eCCy4AYN68eWzbti393pYtW5g/fz52ux2v18vUqVPZsWNHRucTotB0tvl5/Dv/3Wu08PtHniUYCOWxVcNTVlHCQz+7j5XXLKdm+mRuuOUavvyNfxwTo7bRJiOfLAkdPQRGsnehkSR09FBGox+/309RUVH6tcViIR6PY7Va8fv9eL3e9Hsejwe/3z/icwlRiAzDwN8R6FWWiCdIxBJ5atGpqZpUwVf/3/+HcDBCUZEbu1NmJkBGPlnTPeIZbvlwFRUVEQic+IeXTCaxWq39vhcIBHoFIyHGA4fLzvmXLexVNn3ONByusTN15Xa7KKsokcDTgwSfLDHb+v+hGqh8uBYsWMAbb7wBwObNm5k9e3b6vXPOOYf6+noikQidnZ3s3r271/tCjAfuIje3fOUmrr3tKmrVVC6/fjlf/u4d+ErlD62xTKbdssQ1cTKBg/t6T72ZzLgmTs7ouCtXruT111/npptuwjAMvvWtb/HYY48xdepUVqxYwZo1a7j55psxDIO77rpL9iYR41JxmY9PfO5qVv3tCpwuh4wgxgEJPlnSfV8n26vdzGYzd999d6/0OjNmzEh/feONN3LjjTdmdA4hxgKr3YrPLqOd8SLnwUcp9R7Q3vVyL/AT4PtAHFirtb5PKWUGHgbOBSLA57XWu5RSSzKpO9rX5igtzzjYCCHE6SCnwUcp5QTQWl/So2wzcD2wB3heKbUAqAWcWuulXUHkIeBa4MeZ1NVav5uL6xRCCDG4XI98zgXcSqm1Xee+F3BorXcDKKVeBFYAk4C/AGit31JKLVJK+bJQd8DgE4lE+mxbEIvFCIXy/yyBYRgZtSOTLRnGgnA4PK6vLxPSN4OT/hncaPZProNPEPgu8DNgFvAC0Nbj/U7gDMDHiak5gERXWUeGdQfkcDj67FXe0NAwoq0Msm2kWyp0s9ls43qf+uHsM3+6kr4ZnPTP4IbTP/X19SM6dq6Dz05gl9baAHYqpdqBsh7ve0kFI3fX193MpIKJN8O6QgghCkCun/P5HKl7MiilqkkFjoBSaoZSygSsAtYBbwJXddVbAmzVWncA0QzrCiGEKAC5Dj6PAiVKqTeAp0gFo88DvwLeBt7TWm8E/giElVLrgf8A7ur6/BcyrDsmbd26lTVr1vQpf+yxx1i9ejVr1qxhzZo17NmzJw+tE0KIU5fTaTetdRS4uZ+3lpxUL0kqeJz8+bcyqTvaDryj2f7MekKtnbhKvZz18WXUnKcyOuYjjzzC008/jcfTNxHh9u3befDBB5k7d25G5xBCiFyT9DpZcuAdzXu/foVQaycAodZO3vv1Kxx4R2d03KlTp/LQQw/1+9727dv56U9/yqc+9Sl+8pOfZHQeIYTIJQk+WbL9mfUkTto0KhGLs/2Z9Rkdd9WqVelEoidbvXo19957L48//jj19fX89a9/zehcQgiRKxJ8sqR7xDPc8kwZhsGtt95KWVkZdrud5cuX88EHH4zKuYQQItsk+GSJa4AMuwOVZ8rv93P11VcTCAQwDIONGzfKvR8hxJghwSdLzvr4Miy23tNjFpuVsz6+LKvnefbZZ3nqqafwer3cdddd3HLLLdx8883MnDmT5cuXZ/VcQggxWiSrdZZ0r2rL9mo3gMmTJ/Ob3/wGgGuuuSZdft1113HddddlfHwhhMg1CT5ZVHOeykqwEUKI8U6m3YQQQuScBB8hhBA5J8FHCCFEzknwEUIIkXMSfIQQQuScrHYrcIlEgm984xvs378fi8XCAw88wNSpU9Pv/8///A8//OEPsVqtXH/99dx44415bK0QQgyPjHyy6MVn/sonln+WZbNX84nln+XFZzLPtdadr+3JJ5/ki1/8Ig888ED6vVgsxgMPPMDPf/5znnjiCZ566imampoyPqcQQow2GflkyYvP/JV//18/IByOAHD08DH+/X/9AIBVH790xMe9/PLLWbx4MQCHDx+moqIi/d7u3buZOnUqxcXFACxcuJBNmzZx5ZVXjvh8QgiRCzLyyZIfP/R4OvB0C4cj/PihxzM+ttVq5Wtf+xrf/OY3WbVqVbrc7/fj9Z7IHefxePD7/RmfTwghRpsEnyxpPNL/dNdA5afqwQcf5MUXX+See+4hGAwCUFRURCAQSNcJBAK9gpEQQhSqnE+7KaWqgHpgJRAHfgEYwDbgDq11Uin1DWB11/t3aq3fVkrNzLTuaF7XhEmVHD18rN/yTDz99NMcPHiQf/qnf8LlcmEymbBYLADMmDGDffv20dbWhtvtZtOmTdx+++0ZnU8IIXIhpyMfpZQN+AkQ6ir6HnC31voiwARcq5RaACwHFgM3AT/MRt3RvrYvfOVWnE5HrzKn08EXvnJrRse94oor2LFjB5/+9Ke5/fbb+bd/+zfWrl3LU089hc1m4+tf/zq33347N910E9dffz0TJkzI6HxCCJELuR75fBf4MfCvXa8XAq91ff0CcAWggbVaawPYr5SyKqUqs1D3j4M1LBKJ0NDQ0KssFosRCoUG+ERvF69cQjQa5Wc/+BVNR1uonFjO57/4aS5euWTYx+iPyWTi29/+NiaTqc97oVCIpUuXsnTp0l5lJ4vFYn2ubTwJh8Pj+voyIX0zOOmfwY1m/+Qs+CilPgs0aa1fVEp1Bx9TV+AA6ASKAR/Q0uOj3eWZ1h2Uw+Ggrq6uV1lDQwMul2t4Fwhcc8Mqrrlh1dAVT1EoFDqldpzMZrP1ubbxpKGhYVxfXyakbwYn/TO44fRPfX39iI6dy5HP5wBDKXU5MA/4L6Cqx/teoA3o6Pr65PJkhnWFEEIUiJzd89FaX6y1Xq61vgTYDNwCvKCUuqSrypXAOuBNYJVSyqyUmgqYtdbNwHsZ1hVCCFEg8v2Q6VeAR5RSdqAB+J3WOqGUWgdsIBUc78hG3ZxdkRBCiCHlJfh0jX66Le/n/XuBe08q25lpXSGEEIVBHjIVQgiRcxJ8xoDjx4+zfPlydu/e3av8scceY/Xq1axZs4Y1a9awZ8+ePLVQCCFOTb7v+Ywrzz/9Ej/49iMcPXyMidVVfPFf/o7V163M6JixWIxvfvObOJ3OPu9t376dBx98kLlz52Z0DiGEyDUZ+WTJ80+/xH1f/w5HDjViGAZHDjVy39e/w/NPv5TRcR988EH+5m/+hqqqqj7vbd++nZ/+9Kd86lOf4ic/+UlG5xFCiFyS4JMlP/j2I4RDJ2W1DkX4wbcfGfEx//CHP1BWVsayZcv6fX/16tXce++9PP7449TX16f3/hFCiEInwSdL+ksqOlj5cPz+979n/fr13H777TQ0NPC1r30tvVmcYRjceuutlJWVYbfbWb58OR988MGIzyWEELkkwSdLJlb3nRYbrHw4fvWrX/HLX/6SRx99lLq6Oh588EEqK1NZsv1+P1dffTWBQADDMNi4caPc+8mSSGeQtv2NdB5pIeofeV4+IcTAZMFBlnzxX/6O+77+nV5Tb06Xgy/+y99l9TzPPvsswWCQv/3bv+Wuu+7illtuwW63s3TpUpYvl0ebMhVu9/P2D/9IuC21KV9J7STOveUKHEXuPLdMiPFFgk+WdK9qy/Zqt25PPPEEkNrDp9t1113Hddddl5XjC0gmkuxbtyUdeADaPjpC56FmHGpqHlsmxPgjwSeLVl+3MmvBRuReMpEg2NzepzzQ3E6FykODhBjH5J6PEF2sdhtTzu+dPt5kNlE5R0Y9QmSbjHyGYBhGvxu5jRWGYQxdSaSV1E7krBsvZd/r72Ox25i9eil2ud8jRNZJ8BmE0+mkpaWF8vLyMRmADMOgpaWl3+wIon82t5PqBSo12jGZsXuk74QYDRJ8BjFlyhQOHjyYfrYmX2KxGDabbUSfdTqdTJkyJcstGt9MZpOMdoQYZRJ8BmGz2Zg+fXq+myFb/Qohxh1ZcCCEECLncjryUUpZgEcABSSA2wAT8AvAALYBd2itk0qpbwCrgThwp9b6baXUzEzr5upahRBCDCzXI59rALTWFwD/G/he1//u1lpfRCoQXauUWkBqJ9LFwE3AD7s+n1Hd0b88MR4konGi/hDJeCLfTRFi3Mpp8NFaPw38fdfLaUAjsBB4ravsBeBy4EJgrdba0FrvB6xKqcos1BViUOF2Pzv+tI5NjzzLnlfqJbebEKMk5wsOtNZxpdTjwCeAG4CrtdbdD6N0AsWAD2jp8bHuclOGdQcUiURoaGgY8XWNpnA4XLBtKwTZ6p+J5ZV88MTLBBpbAfAfaSHY2smEi87keEdbxsfPB/nZGZz0z+BGs3/ystpNa32rUuprwEbA1eMtL9AGdHR9fXJ5MsO6A3I4HAW7okxWuw0uW/0TbGlPB55ujZt3oa5awoTJkzI+fj7Iz87gpH8GN5z+qa+vH9GxczrtppRao5T6166XQVIBYpNS6pKusiuBdcCbwCqllFkpNRUwa62bgfcyrCvEgMxWCyZz74eJ7V4XZPEB43BHkKbdhzmweRfBNj/JhNxXEqenXI98/gA8ppR6HbABdwINwCNKKXvX17/TWieUUuuADaQC5B1dn/9KJnVzcoVizLI67Ey/bCF7Xt6UKjCZOPOTy7OW5SDcEeT1h5/h2IeHALA57Vz5vz+Db0JpVo4vxFiS0+CjtQ4AN/bzVp+NaLTW9wL3nlS2M9O6QgzE6rQz7cKzmTRvJoHmdryTyrG5nZjM2Zkg6GhsTQcegFg4yvt/fIMlt63C5rBn5RxCjBWS4UCIHmxuJza3E09V9kcjofZAv2XJeAIcWT+dEAVNgo847cSCYWKhCPFgBEdJEXZP9kY3g6mcMQmrw0Y8EkuXqcvm4/C4BvmUEOOTBB9xWokFw+x+cSNH39UAWF0OFvz9tbgrSkb93A6vmyvv/jSb//AGoY4gc1bMY2LdtFE/rxCFSIKPOK1E/aF04AGIhyLs/stG6m64FKtzdO+7WKwWiqvLWfb5K0nGEziKZMQjTl8SfMS4EfUHSURimKwWLHYrNlffVWqRzmCfslBrR05T6dhGOcgJMRZI8BFjnslkItIRYNsTfybYlHqWeMK8WUxfuRjbSfdTPJUlmG1WkrF4umzCubOwuuSOvxC5JFsqiDHPV+Tl0Iat6cAD0Lj5Q0KtnX3qWt1O5v/dxymeNhFniZdply5g0kKF2SL/FITIJRn5iDHPajbjP9Lcpzx4rBXflKpeZRarBe+kCubefAXJRBKby4HZaslVU4UQXeTPPTHmheMxKuae0bvQBMW1Ewf8jM3txOF1S+ARIk9k5CPGvEAgwMy66UTa/Bzd1IDV5eCMjy3tc79HCFE4JPiIgmQYBrFAiEh7AIvDls48MBCbx8XU5QuYvHgumMDmdvVJEiqEKBwSfERBinYE2PqLZ4kFUpu5lcyYwsyrLxp0NGOxWbHY5EdaiLFA7vmIgpOIxTnwxuZ04AFo232QcGtHHlslhMgmCT6i4BjxBJF+Ak24re/SaSHE2CTBRxQci9NO5bmzepWZzGZ8NQOvXhNCjC0yQS4KjslkonRGDdMuX0zju6nVa7WXLx50wYEQYmyR4CPyJhGLYyQSWJ19U9vY3E4mLqyj4szpmMxmCTxCjDM5Cz5KKRvwc6CW1NZZ9wMfAL8ADGAbcIfWOqmU+gawGogDd2qt31ZKzcy0bo4uVQzBMAyinQEaN24h6g9QPncW3ppJfYKQ2WLGXuTOUyuFEKMpl/d8PgO0aK0vAq4E/hP4HnB3V5kJuFYptYDU9teLgZuAH3Z9PqO6Obg+MUzxYJgPf/MCrXovgUPH2P/im7TvPYhhGPlumhAiR3IZfH4L3NPjdRxYCLzW9foF4HLgQmCt1trQWu8HrEqpyizUFQUifLydRDjaq6zlfU0iHMlTi4QQuZazaTettR9AKeUFfgfcDXxXa939524nUAz4gJYeH+0uN2VYd1CRSISGhoYRXNnoC4fDBdk2k8lEUVERJpMJv99PMjn0zKbFYmFiUd9dQy1OB63tbTR/1HrK7SjU/ikE0jeDk/4Z3Gj2T04XHCilaoA/Ag9rrX+tlPp2j7e9QBvQ0fX1yeXJDOsOyuFwUFdXN/yLyaGGhoaCa5uRSBILhmjZ9iFGMknt3FnYPMNL1BkPhfFMnkDgUCMAJouZ6gsW4KospXLiqS+nLsT+KRTSN4OT/hnccPqnvr5+RMfO2bSbUmoCsBb4mtb6513F7ymlLun6+kpgHfAmsEopZVZKTQXMWuvmLNQVWRQLhtC/fo5j9dtpeq8B/evniAX67hIKqWATaeskfLydWDCE1eVk6uVLqb1qOdUXLWTm9VdgtlszvucTC4Zp23OIvS+/TdveQ8SC4YyOJ4QYPbkc+fwbUArco5TqvvfzJeAHSik70AD8TmudUEqtAzaQCo53dNX9CvDISOuO/uWdXtp27uu1G6iRSNL0vmbyhQswmU/8TRMLhjnwygY69x0GwFlWzPSPX8be514jFgxjdTo49Fo9Nq+H2TeuGnEm6ngkyoF1mzny9nYADm/YSvXSs6m5eD5Wuy2DKxVCjIZc3vP5Eqlgc7Ll/dS9F7j3pLKdmdYVWdTPmNlk6v6/E0JNLenAA6nFBsc/2IXd5yHY2EKsMwBAtL2TZCIx4uYkIjGObvqgV9mRjdupPv8sCT5CFCBJryNGpGRWLeYev9RNVgsV58zBdHLwae57uy3c3IajrPeiA4vTgdk89I/jYFNzJ78lS7eFKFyS4UCMiM3tRN28mtYdezESCcrqZmDtZ8rMVzuZoxs29yormTUNZ1UpzZt3kIhEMdttTL/qIiyuvpkOusUCIVoa9hJqOk7FObNxlRf3eijVYrdRefZMmrZ8mC6bMG82Fhn1CFGQJPiIETGZzdiLPExYNHfQehaHjZoVSzhWv51kPEH5WTMxMDBbrdTdcg2JaDy1D4/LgdnSe6WcYRgkwhEMw2Dnb9cSPJZaht28dRdnXH0xZXXT0yMtq9PO9MvPo3TGZFp3HaB01lRKpldjddpHpwOEEBkZdJ5DKTVLKfV7pdQvlVKzepT/aPSbJvItEY1ldB8GwEgatO78iKpFZ1N9wQLCx9vZ/9IGTIDN48ZZ6sNW5O4TeOLRKJF2P8f1Plp37mPaFcvw1Van3z+8/n3iJ61ms3lcVM6dwayPX0zlWWdIPjghCthQI5+fAg8ANuBppdRntNbvAXNGvWUibxKRKKFjzbTrPViLXJTNnYPV4+5zP2c4zFYr8WCYAy+vT5f5zpiC2Tr4j14iHGXnb18k2pFakGC2WZl9w0qCjS3EQ5FBt8g2DePekRAiv4acdtNarwVQSu0C/qCU+hiphJ1iHDIMg8DhozS+8U66LNx0nMkrL+43+/RQbG4nZ1xzKUfe2kywsQXv1ElULTwLi2Pw6bCOfUfSgQcgGYvTtOVDSmdPo+n9nUy+cD5WGdkIMWYNFXziSqlrgD9rrbVS6p+A50iNhMQ4lIhEafsgddPe4nIyYelCjHiCcFMLzvJSLC7nKY+AbEVuJi8/n2QshsVuH1YWhJ7PEKXL4nFKZ02jat4c7CVFIxqJCSEKw1DzE58DrqcrN5rW+q/AnUB0sA+JsctkMmG2p0YlEy84j5b3tnHk9bc48uoG9j/3MvFgaETHtdis2NyuYQUegJKZNZhtvf82mrDoLMpULe4JZViHGDkJIQrboMFHa30A+LLW+jiAUqoK2Kq1npeLxoncszjsVCyci6OshFggQKS1Pf1eIhKldZsmmRj9rZFsHhdnrrmGirmzKJ09jTk3X4WrzDfq5xVC5MZQq92Wk8qTVtpVdC5Qr5S6cNRbJvLG7vNRfekF/U59xYMhGEb26kyZLRacpT6mXr6Y2o9dSNGkSix2Ge0IMV4MNe12P7Bca90KoLV+CVhJagWcGKfMVgsmqwVPTXWflWMlc2b2mQ4b1bZYLFhyeD4hRG4MueBAa/1RzwKt9U6llGxJPY4lwhGOb9tBPBKh+rJltH7wIclYnNKzZuMo77sXjxBCnKqhgo9ZKWXWWqeDjVLKAsj8xzgWaWunfeduAKLH2yiaNgX3pAm4Ksvz3DIhxHgxVPD5FfDfSqlvAXuAGlJbYT812g0T+RM+1oJv5nSsLiehY820bttBrNOPo7x0WMk/hRBiKIMGH631T5VS7cD3gGpgH/CY1lqCzzjmnV5D6wc7CTc145k8ieJZ0zGShgQeIUTWDBp8lFJ3AF8FEsA/aa3/kpNWibyJh8Ic/p830s/zhBqbKT17DsUzz8hzy4QQ48lQf8p+GlDAEvrfCE6MUclEst/9bhLhSJ8HSTt375O9cYQQWTXUPZ+w1joKNHdtSZ0xpdRi4EGt9SVKqZnAL0jlitsG3KG1TiqlvgGsBuLAnVrrt7NRNxvtH+vi4TCx9g78+w9iL/ZRNLUGS4+cbf1lIDA77H12KBVCiEycyiR+xr99lFL/AvwM6M4I+T3gbq31RV3Hv1YptYDUFtiLgZuAH2ajbqZtH+uMZJJYIEDgwEGObXib4KHDtH2wg8b1b5GIRNL1zDYb7imTTnzQZKJi/tkjSioqhBADGWrkc5ZS6tekfoF3fw2A1vrmEZxvN/BJ4Imu1wuB17q+fgG4AtDAWq21AexXSlmVUpVZqPvHEbR33EhEo0Tb2uncs69Xeayjk0Q0isWRCi4Wp4PK8+YTnzOLaKcfV1VFr5GREEJkw1DB58YeX/8405NprX+vlKrtUWTqChwAnaQSmPqAlh51usszrTuoSCRCQ0PDqV1QjoTD4YzbVjtxIkYi2e9eN4lEss/xrVYrVquVyMEDBX+/Jxv9M15J3wxO+mdwo9k/Qy21fm2w97Og530YL9AGdHR9fXJ5pnUH5XA4qKurO5W250xDQ0PGbYuHQnQ07MQ3awbHN29JlzsqK7C7nAV77cORjf4Zr6RvBif9M7jh9E99ff2Ijp3vBzfeU0pd0vX1lcA64E1glVLKrJSaCpi11s1ZqHtaM1mtuKoqSYRCVC1bjG/mDCrOW0DlovnpKTchhMiVfGds/ArwSNdKugbgd1rrhFJqHbCBVHC8Ixt1c3ZFBcpis1E0rYZkPE4ynsA3ewZmq1W2nBZC5EXOg09XotIlXV/vJLVa7eQ69wL3nlSWcd3Tndlmw2yTTWiFEPknf/YKIYTIOQk+Qgghci7f93zEMLhcLuLhMIGDByGRwFNTg9nhwGzpm41ACCHGAgk+Y8DEigqOrV+PEU9ta+3fv5+qZcswezx5bpkQQoyMTLuNAdHm5nTgAcAw8O/di5GUdHVCiLFJgs9Y0F9ST0n0KYQYwyT4jAH28vLeS6TNZry1tfKMjhBizJJ7PmPA0aYmapYuJXjkCMl4HM/kyZglK4EQYgyT4DMGhMJhLE4n3unT890UIYTICpm3EUIIkXMy8hFCFKT21g462v2EgiEqqsooqyjNd5NEFknwEUIUnLbj7Xzv3h+z7qW3AKicWM5//voBqiZV5rllIltk2k2MG/FIjFB7gGggnO+miAwdPtCYDjwATUdb+K8f/ZZIJJrHVolskpGPGPPMZjPhjiANz7/FkS17KJpQyvybLqWoqoRoMEIiGsdkNmFzO7DaJav3WHB4/9E+ZQf2HiIajuJw2PPQIpFtEnzEmOcr8rL9mTfZ/1Zqu99IZ5A3H/4Ty7/8N7z9i7/QtPMQFruVs6+7kKnnz8HulmXqhW7uwjlYrBYS8US6bNV1l1Lkk5RS44VMu4kxz2l1cOT93b3KqlQN+qV6mnYeAiARjbP5N68S8Qfz0URxikpKi/mPX3yT2WfNYOLkKv7+q7dwwWXnY5LMHuOGjHzEmJdIxvFUltC2/1i6zDupnH0bG/rUbT/cgrdKVk0VOqfLwdkL63jwp/eQTCTxlXqxWuXX1Xgybr+bSikz8DBwLhABPq+13pXfVonR0BEKMP9Tl7Hu+38gHo6CCTwVPipVDW0HmnrVLZlckadWipEoKSvOdxPEKBm3wQe4DnBqrZcqpZYADwHX5rlNYhREIhG8U6ex8p41hNoDOIqcWF0Oymon0XGkhRTqnCEAACAASURBVMbt+7A67ZzzyYuwF7ny3VwhBGAyDCPfbRgVSqnvAW9rrZ/sen1Iaz15oPqbN282HAWaLy0cDuN0OnNyLpfLRUVxCaZEEqwWWjs76PT7c3Lukerun1JfMR6nCwyIxGO0BzrxuopwWG0YGPgjwYK/lmzL5c/OWCT9M7jh9E8wGKxfuHDholM99nge+fiA9h6vE0opq9Y63l9lh8NBXV1dblp2ihoaGnLWtlDTcT760yskwlFMVgs1K5ZRPWsW5gKeb29oaGDmtOnsf+N9GtZtwUgmKZs1hTNvuKzXSKeE8jy2Mj9y+bMzFkn/DG44/VNfXz+iY4/n1W4dgLfHa/NAgUekxINhDryynkQ49SCfEU9w8JUNJAr8wT6r1UqotZN9r21Ob7B3/MODHN7UQDIhG+4JUYjGc/B5E7gKoOuez9b8NqewJeNxkokE0bbOvuU9nrUoRDabjY4DjX3K2/YeIRmTvzdE9sUL/N/EWFC4cymZ+yOwUim1HjABt+W5PQUtcrydcMtx3JMqCR45sULM6nZhthX2j0k0GqWidlKf8nI1FYu9sNsuxpbjLW28uvZN3n17C1dcfQnnzD9TVuSN0Lj9l6m1TgJfyHc7uiXjcTCM3juSFoh4OELTpveJBUNMungxjRu3EDjUiLO8lCkrlmB1FuZCjG6JRAJHcRGzVi9jz0vvkIzHqTpnJhPOnSm7vYqsaWtt5+47v8Vbb6Tucfz56Zf5+y/dwm1f+BQOp6T8OVXjNvjkSjKRgGQSk8XS7y+6ZDJBIhQieHAfRjKJa2I1Nm9xYd3ANwwSsRjxQJAjr79Fad1sqhbNxepy4SjxDv35AmBzOag+r46qs88AAyx2G1b5hSCyKBgIpQNPtyce+Q03fPoaHM6yPLVq7JI/CzOQjEYJHt5P596dhBoPk4zF+tQxojHaG7YS62gn7u+kc5cmHgzkobUDszgclNbNAiAeCNG06X2OrNuIxVF4o7TBWGxWHF4PDp9HAo/IOpO5b2ofi8UC4/NplVFXQH9+jy3JWIzOjz4kEQ4BkAiHSEajeKZMw2SxpOtF247DSc9ShY8dweopwtyjXj6ZzCaKpk3GbLfT/uEe7N4iys6eg6XAp9uEyCW328XK1Zfw0vOvpsv+4Uu3UFw6NmYHCo0EnxEyksl04OkWbT+Ou3pKr+Bjtvf9BW6xO3slSEzGYhjJBGDCZLHkZUrO6nDgq52Cp3oCJou5YAKjEIWiuMTH1+77Z1Zfdzmb67dxycplTDujBlsB3scdCyT4jJTZRGoR3YlRjclqA0wkY1GSiQQmkxlrUREWl5tEKJiu45w4KX1/KBmL0rl7F5GWZgCcVRMpqq3FbMvPtJGla7+bZDyOyWyWG/ZC9FBWXsLFly/l4suX5rspY54EnxEymS24Jkwi1Hg4XeaZPA3DMGjX20lGUrtpuqtr8M2qIxmNYCSTWFxuDFL3i0xWK5HW1nTgAQgfO4qjvAJHWX5uYCYiEUJHm+jcdxBHWTG+GdOxuiT9iBAiuyT4jJDZYsFRXoW9pIxEJIzF6QKTmcCBj9KBByB4+AA2rw+LwwkmE+HmYwQPHgAMXNWTsXl6zBebTJjMZqLtbSMKPsl4HEymEU+ZJRMJ2j/cS+vW1FYEwUNHCBw4QvWlF8j9HyFEVknwyYDZagWrNRVYgEQsRqKflWyJcBiTzUYyFiOwb2+6PHhgP77ZdVg8RbgmTMTq9mAk4qnRkWEMe+OsRCxGvLMT/0cfYbJa8c2YgcXl6jNllozFScZixDo7sRZ5sNjsmHtsK52MxmjXvTdli7a1k4zF8hp84pEoyWgqU4HFaccyjIde4+EIRiKJ1e0s6A3IkskkRtLAYpV7bOL0IsEni8wWC/aSUkJHey9E6A4EPafXTGYLRjJB5HgLvpmzCRzYR+eunanj2OyUzZufGk0NQ7yzk+Z33km/Dh87xoQLL8TSIxutkUwSaWmh5d330mW+2bMoqp3Wa4GD2WoheXIqtzz+8o4FQ+z/az3N23ZhtlqoXnYuVfMUNlf/wTARTxBuaWPP2reJh8JUL55L2ewabAU4dRhs9bPz1c34m9qZfdk8SqrLsbsLr51CjAa5m5xFJrMZ14RqHOWVqekvuwPvGbNIRKOYzBbsPh/28gqKzzwH76w5FJ95Do7KSsAg0nwipU0yFqXzo70kE0PnJUvG4/g/+qhXmZFIEG5u7l0vFqN12/ZeZR0f7sKInziHxeGgfN7cXnXcU6qHnV7HMAziwRDRTj+xYAgjkVn+K8MwOL5zP81bPwTDIBmLc/C1eiIn5Z/rKR4IsfmRP9G2+yD+w83s/OOrtO05PGD9fAm1B/jL//kV257byEcbd7D2gSc59uGhfDdLiJyRkU+WmW02PDXTcFfXYCSTmCxmTObU8mmr14fLYkvf2zFZLJAEo5/My4lQKJWheajZGJMJUz9LPc32k1bLGUafh2BNZjNG0iDa0UkyGsVa5ME1sYqaq1YQONyIo7QYe0kxliH2OTIMg0QkSjwY4sDadcQDQSxOB1NWXICrsqzX0vNTkYwnaNt1oE95x74jFE3qf0fS9n1H+yRCPbKpgdIzJmMdYLSUD+2HWwi29t5baNvzb1MxoxqnbHgnTgMSfEaB2Wrrt2eNpIGRTBLv7CR46BAmiwXvjBlY3G4wmbCXleOaMBGT2XLiOEOdy2LBN2MG4cbG9EjD6vFgL+6d7NBkseCaOIHQkaPpstJzz+H41u3pMrPNxsTlF2Av9mEv9g37emOdfqJtHTS+/T7xQGpJeSIc4eDLbzD9kx/D5h7ZL1Oz1UJxbXWfAOSdUjXgZ+w+T58yR7FnxAFwtJhtfdtjsVkK+v6UENkk0265ZBgkIhHaGxqIdXQQbW2lpb4eEgnK5p+H3VdCy6b3OPbmBjp37yUZHd4+OhankwkXXkjJ2WdTNn8+Feed12e0YrbZKDmzjqIzpmPzFlE0bRoWp7NXMEpNzTWQ6CdN0EAS4QjHNmzCbLcTbe89HZaIRDEySD1vMpkoP3M6JTNrUtdgNTNpydk4B8ki7K4swTdtYvq11eVg2vIFBZfd2ltVQsnkE6M3k9nEvE9eiMMj93zE6aGw/kWeBsJHj/YuMAzCTU04Kys5vnlLujh09CjWIg++2bOGPKbJbMbidOKprh60nsXhwDd7Fsb02tQzRs0tferEQyFIDn8DNsMwiBxvIx4I4CgrIXK8Lf1e6vkgg3goPOJnhWxuFzNWX0QiFsNIJEnG4ukN4/pj97iou/FyIm1+4uEInqoybAX4C93l87DiKzfQuOMA/uZ2pi6ajaukKN/NEiJnJPjkkMlkwtzP/ROL00mkra1PefhYE97ptVltg9lshq422Hw+TBZLr4UBRdNq+t4v6kcykcBIJDBZzDirKmjbsYtJyxZy5M1NRFrbsXk9TLxgEYfXvZNapbZ8+FszJKKpkVd3tgXDMDj0xnu0bEstA7cVuan79FU4ivv/ZW33uLB7MrtvEg2ESSYSWJ12rPbRSZ/iKvZQu3jOqBxbiEInwSeHLA4HRbW1hI8dS0+p2Xw+LE4XpnjfqS5HaSmmUczzZrGn7vG0bv2ARDhM0bSpeCZPHvK+QzwcpuPD3cTa23FNmsCEZefR+OZGmjZtZtLF54NhEPMHaXp3G5Hm46nPBENDBp9ELEaktYOm+m2AiapFc3GU+Ih2BtOBByDmD3LojfeYdsXSYT3z0y0ZTxALhgm3dmL3urG6HH2WbBvJJIGmdrb97lX8x1qZcNZ0Zl+5GIfXPezzCCGGJsEnx0xWKxXnn088EMBkNmN1uwk2poKRd+YMOvfshWQSe0kxRdNrM16uPGhbLKnl3xXnLYCkgdluGzKXWyISoWnjO8T9qZVa0fZ24sEQEy5cDEkDI2lw8OV1RFrbT/rc0PevYp1Bjm3altqq4mgznfsPM/tTVxNp77u0OtzSTjIWP6XgEzh2nM2PPpveWrvm4nlMveDcXqvgov4Qbz38R6L+1LNaBzZ+QDKR4KxPLsc6xraYEKKQ5Tz4KKU+AfyN1vrmrtdLgO8DcWCt1vo+pZQZeBg4F4gAn9da78q0bm6vtH8Wmw1sNsx2O6auJdfuCVU0rt+Iq6qCysXnYzKbMNtstNRvwjtzJm736P7VbRnGNFs3Ix5PB55ugYOHKJ41C4vbiZFMUlI3k8b1JzbdMjvsOIoHTzufjKfu5bjKS7A47ExYPI9Dr27keMMuSutmdi0LP3Gvp+zMM05pz55oIMSHz7yRDjwAB17fTPWiul7BJxaKpANPt8ate1Crl0rwESKLchp8lFLfB1YBm3sU/xi4HtgDPK+UWgDUAk6t9dKuIPIQcG2mdbXW747+VQ5Pz/xrFqeTCRcsIeYPYCTiGAlo3bKFRDhM6OhRXJMn57GlJzH3s0TYbk8l+Ca1+ME3bQpmi4U2vQdrkZvKBXOxDLHgINruZ8/vX0wHGLuviMmXLibU3IbN5UTdtIr9L28kFgxRcfYsys8845QybhtJg3A/D6fGI6kM5KlLs2Bx2FMZHXrsweSuLJHs3kJkWa5HPuuBp4F/AFBK+QCH1qmEYkqpF4EVwCTgLwBa67eUUouyVHfA4BOJRGhoaBiFSx4em83GpJIS2rdtw+ix1Nnq9dIWCuW1bT1VlpXhqZ1K4KP96TJf3RwOHT1KIBhMl7ndbjwLzyKRTPLRkcMkBpk+LCspIfz+h71GNtEOP9HOAJ7pk9G7PsTpdDLpyiVYTBYC0TC7euTIC4fDQ/ZPia+YirOmc+SdE/VsHidWh529L24AA6oW1RHDYPaV57PzhY1ggNVh48xPXsyhY0eIHIicUl8VguH0zelM+mdwo9k/oxJ8lFK3A3edVHyb1voppdQlPcp8QEeP153AGV3lPW8aJLJUd0AOh4O6urrBqoy6RDRKtLKS4OFUOhib14t32jSO7t2b97b1lCgtpahmKjG/H0dJMWabjaJBNtSawKTBjxeNciCxo0+52WrF5S3qc+3FQDUnRoMNDQ3D6p/yklIsdhvN2/fiqihm5pVL2f38OvwHGgFo2b6bcz7/CbzLzmbyAkU0EMLh82DzOCkrsIdUh2u4fXO6kv4Z3HD6p76+ftD3BzIqwUdr/Sjw6DCqdgA9bwZ4gTbAfVK5OUt1C5rFbsc3Zw7emTPBMDBZLFgcDgyjsDaJt9jtWOx27L7sbB9ssdupXHAW/v1HTpQ5HbgnVmbl+N3sHhfTL1tEzbJzMFktNG3ZmQ48kFrp1vjeDqatOB+by4FrnG6PnIjHiXSGMTCwOezY3YWTdkicPvK62k1r3aGUiiqlZpC6N7MKuA+YAlwD/KbrPs7WLNUteN0LEk43zvJSZtzwMVq2aKxuJ2Vnz8Y6ChmezTYr9u4Vcsm+Qd0ySs/0FIpoMMKejTt469d/JRaOMmPJHJauWYGrn7REQoymQlhq/QXgV6RSaK7VWm9USr0DrFRKrSd1K/u2bNTN2RWJU2ax23BVllF9yWJMJnJyg79i7gyOvL2NeCh1L8fitFN57uxRP28+Bdv8vP6zv6Rf71rfQGlNJeeuPn/EmxAKMRI5Dz5a61eBV3u8fgtYclKdJKngcfJnM6orCp/ZkrtVZbYiN2d/7lpadx0AA0pn1WDLMDNCLiTicaKBCCaTCafv1JbhN+7qu73Egc17OPOyeTgkm7bIoUIY+QiRFyaTCbvXw4T5hZnixkgaRPxBDMPAYrNidzsJd4b48NXN7Hp9Kw6vi0U3XUpZ7YRhpwCqqJ3Qp2zSnJpTemZKiGyQhxdETiTjCSLtnRzZ8B6Nb28h2hkYNEHo6S4Ri9O86xCvffcpXrz757zzi78Q8QfZv0mz9ZkNhNr8tB1o4pXv/Y5IZ2joA3bxlHlZ8Mll6RHmxDlTOOuKhbKNt8g5GfmInIj5g3z41PPpjfOat2hm3bQae5HkTOtPNBhmw4//RDKWej6qqWE/TR8e4qO3ej9zYSSSNO89iqd8ePsvOYtcnHPVYuounUcymcTmsOGUvHUiD2TkI0adkUzSvGVHrx1bk9EYHbv2D/Kp01ukPZAOPN3aDzThm1jWp663cuD9jfpjd9nxlHnxVhRL4BF5I8FH5Eg/mbJl084BObweTOYTHWQvclE0oZS51yzBXXbi+aPaJXPwlA1/11khCoVMu4lRZzKbqThX0dqwO52l2+ywUTxjap5bVrhsLjvzbl7BlqdexV3uY/5nVqJffpcD733I8juuBZMJm9OGzeXos0rNSBqEO4MYiSRVpRUDnEGI/JLgI3LC5nEz+1OrOd6wG5PFQumc6aPyEOl4YXXamTx/FlVqKslEkte+/wf8x1JJOg5v3sP0pWcy728vwXbSKrVkIkHbgSY2PPJnYsEwMy+bh2OJDXeZL6fL2IUYivw0ipwwWy3YfUVMXHwuExbNxV7kkUzRQ7DabbhKijCSyXTg6fbRxh3EI303IIz4w7zx8DOYTHDhP38C//EAr//4ObY+9xbhjmCf+qMl7A/R2dxOe2MrwfZAzs4rxg4Z+YiClkwkiIcixEMRrE47Frstte3BacRstWIymzB6pANylRbR34aziWiMqD/EObdewcYnXqblo1TuuuY9R+lsbOX8NZdjd41uLrdQR5ANT77K1pdSSeQrplZx3d2fwjNOc+WJkZE/PUVBCzW1svVnT7Pt539i849+R/MHe0hE+/7F31MiFifaGSQeHnr31LHA5rRz1jVL069NZjOLPr0CRz/L1C12G3aPE3eZLx14uu0dYLSUbR1NbenAA9C8/xjvPruReI+N/PoT7AgSaPOTTMjzX6cDGfmIghULhNjz5zdPbMFtGOx/+W1KZ9YMmAA06g+y/9V3ad19CM/Ecs5YtQRnSVEOW519NpedmcvPoWbhbAIt7fgmlmF3O3uthutm9zhZ9g9Xg9mMyWLutbzd4XFhysESw+b9x/qUNe4+QjwSw9rPtufRcJTGPUd49RcvEQ1FWXTNEmYvrcPllXQ/45kEH1GwDMMg3NLeuyyZ7LUVdk/xcIRdf17P8R37AIi0+wk2tXLOZ6/GPsbzltndTuxuJ96qkkHrWawWyqZPJNwZYu5V57P12bfS751386XYvae2yCPYFuD4wSaMpEH51Ercwwjkk+dMTS2j75E0fNbSOhwDLDAJtvl56p7/Sk8rrv3Rc7hLPMw6X51SW8XYIsFHFCyzzUrJzBpad+5Ll9k8LswDjHqSsQTHde8HV8PHO0jEYsDYDj6nwmyx4C4p4sxVC6lZOBN/YxvltRNxeF1YTiFzdbDNz5/u+yUdjanFDp4yL5/45q14SgcPQK5iD6u/cgOvPbaWSCDE3MsXMHtZXb8jNYC97+3udT8LYMtL7zLt7OnYXafX/b3TiQQfUbCsDjvTVi7GZDHTvvsgrspSpl+5DNtAS7RN4PB5iLT7TxRZzKftVgEOj4uPDu5HnT+yxKl739mZDjwAgeOd7Hx9C/OvXTb4ed0Ozlg0m0lqChip11bHwIlPS/rJ2lBWXY7Fdnp+304XEnxEQbMXualdtZRkLI7JbB448JAaFc265kK2/3ptOmnp9MvPP60zNiczSN7qb+noU9bZ1IFhGJj6W2rXg9lixjPMe20TzphIzdxpHNiWGuF6K3ws+vgSSXY6zknwEQXP6rDDMJZXm0wmvDUTWPTFGwm3deLwebA6HeN+d9LRMvuis3n/2Y29tnE/c+X8IQPPqXIXe/j4V28g0B4gHo7hqywecmpPjH05Cz5KqWLgl4APsANf1lpv6Nr6+vtAnNSOo/cppczAw8C5QAT4vNZ6V6Z1c3WtIn8sNisWmxWHbAudsaJyL9fe+xne+e06jKTBwusvwDfEgoeRchd7cBcX7vcsmUxiHmcPRcdiMY63tPHBFk1FVTmTayZRVj4639/+5HLk82XgFa31/6eUUsB/AwuAHwPXA3uA55VSC4BawKm1XtoVRB4Crs20rtb6xMMHQohB2Zx2JsyazMovXZe6d1N0+qVDam/p4J1X6jm0+zAXXL2USbUT8YyTTOD79h7k0x//AqFQGIClFy3ige/fk7MAlMvg8x+kRibd5w0rpXyAQ2u9G0Ap9SKwApgE/AVS22ErpRZlqa4EHyFOkcNz+gUdgI7jHfzHnf/J4b1HAHjz+bf4/L23suiyBWN+FOTvDPC9b/0oHXgANqzbROORY2M7+CilbgfuOqn4Nq31O0qpiaSm3+4kNQXX865mJ3BGV3nPBzwSWao7oEgkQkNDw2BV8iYcDhds2wqB9M/AxlvfeD1FOG2pYBiKhfEH/EN8YnAD9Y/ZbMZuONOBp9vzj7/IlNnVtAfa+3xmLHE53bQ0tfYpb25qYdeuXcRiqUwYo/nzMyrBR2v9KPDoyeVKqbOBJ4Gvaq1f6xqh9Ez45AXaAPdJ5WZSwSTTugNyOBzU1dUNfmF50tDQULBtKwTSPwMbT30T9ofY8fo23vnDG2CC86+/CHXhWTgzeIB4sP45tPtwnzKz2YzX66V6avWIz1kIkskkN665lvu+9p10mddXhDpzFlUTTmzDMZyfn/r6+hG1IWdjR6XUmcBvgZu11i8AaK07gKhSaoZSygSsAtYBbwJXdX1uCbA1S3WFEGNU875jvPbYiwTbAwTbArz66F84frB51M7nLfMyVdX0Krv286spGuPpmiAVRFesupj7H/pXzllwFlesvoRfP/MTyitKc9aGXN7zeQBwAt9PrTegXWt9LfAF4FeAhdSqtI1KqXeAlUqp9aQSddzWdYyM6ubiIoUQo0Ov29a37M3tVM+p6ad25nylXr74nS+wfWMDh/YcYfGq8yifWJb1peb5UlLq45rrV3HxiqXY7DY8ntwupMhZ8OkKNP2VvwUsOaksSSp4ZLWuEGLsmqgms+2V93qVTZo9eVTP6SvzsfTKxaN6jnwymUyUlBbn5dxje8mGEOK0MX3BLKacOS39umZuLVPPGXQdkShgkuFACDEmuIs9XPWV64mGImAyYXfacfnGxzM3pyMJPkKIMcPlc0vAGSdk2k0IIUTOSfARQgiRcxJ8hCggiXiCRCyR72YIMerkno8QBSARi+Nv6WDbn98mGU8w96rFeCuLB92ETYixTIKPEAUg2BbgmXseJxGLA7DnrQauvf82SqrL89wyIUaHTLsJUQD2bPggHXgAjKRBw8vvZrQTqRCFTIKPEAXA5uq7U6vNaR83qVyEOJkEHyEKQO0ihavHTp52t4M5K7K/ZbUQhULu+QhRAFwlHq657xYObtlLMp6gZv4MXLIVuBjHJPgIUQBMJhPukiJmX3x2vpsiRE7ItJsQQoick+AjhBAi5yT4CCGEyDkJPkIIIXJOgo8QQoick+AjhBAi50yGYeS7DQWhvr6+CdiX73YIIcQYM23hwoWVp/ohCT5CCCFyTqbdhBBC5JwEHyGEEDknwUcIIUTOSfARQgiRcxJ8hBBC5JwEHyGEEDknWyrkkFKqGPgl4APswJe11huUUkuA7wNxYK3W+j6llBl4GDgXiACf11rvyrRuTi94FA10zfltVfYppWzAz4FawAHcD3wA/AIwgG3AHVrrpFLqG8BqUt/vO7XWbyulZmZaN0eXmhGlVBVQD6wkdU2/QPoHAKXUvwIfJ/U752HgNQqgf2Tkk1tfBl7RWi8HPgv8sKv8x8DNwIXAYqXUAuA6wKm1Xgp8HXgoS3XHi4Guebz5DNCitb4IuBL4T+B7wN1dZSbg2q7v7XJgMXATJ362Mqqbg+vLWFeA/gkQ6iqS/umilLoEWAZcQOqaaiiQ/pHgk1v/QeofCaRGnWGllA9waK13a60N4EVgBamA8RcArfVbwKIs1R0v+lxzfpszan4L3NPjdRxYSOqvV4AXgMtJ9cdarbWhtd4PWJVSlVmoOxZ8l9QfWoe7Xkv/nLAK2Ar8EXgWeI4C6R+ZdhslSqnbgbtOKr5Na/2OUmoiqem3O0lNwXX0qNMJnNFV3t6jPJGluuNFn2tWSlm11vF8NWg0aK39AEopL/A74G7gu11/UEDq+1pMqj9aeny0u9yUYd2CppT6LNCktX6xa3oJMr/mcdM/QAUwDbgamA48A5gLoX8k+IwSrfWjwKMnlyulzgaeBL6qtX6ta4Ti7VHFC7QB7pPKzaSCSaZ1x4uTr8883gJPN6VUDam/XB/WWv9aKfXtHm93f18H+n4nM6xb6D4HGEqpy4F5wH8BVT3eP937pwXYobWOAlopFSY19dYtb/0j0245pJQ6k9Q0ys1a6xcAtNYdQFQpNUMpZSI1TF4HvAlc1fW5JcDWLNUdL/pcc36bMzqUUhOAtcDXtNY/7yp+r2suH1L3gbp/BlYppcxKqamkgnFzFuoWNK31xVrr5VrrS4DNwC3AC9I/aW8AH1NKmZRS1YAHeKUQ+kdGPrn1AOAEvq+UAmjXWl8LfAH4FWAhNZe6USn1DrBSKbWe1M2727qOkVHdXFxkjvyR/q95vPk3oBS4RynVfe/nS8APlFJ2oAH4ndY6oZRaB2wg9UflHV11vwI8MtK6o395oyKjax5P/aO1fk4pdTHwNieuZS8F0D+S1VoIIUTOybSbEEKInJPgI4QQIuck+AghhMg5CT5CCCFyToKPEEKInJOl1kLkWdezEb8hlTDUAFzAr7TW/79S6u9J5XdLAjbgf2mtX+3x2TuBiVrrr+e63UJkQoKPEIXhf7TWNwEopRyknkZvIpWleYXWOqaUmg68rpSaDwSAR0gld/x9vhotxEjJtJsQhcdLKj/fPwDf0lrHALTWe4F5XU+TO0mlkvk/eWulEBmQkY8QheEypdSrpKbXYsA/k8qCvqdnJa11S9d/W4G1XYk1hRhzJPgIURjS027duu7n1NAje7dS6gpgi9b6aI7bJ0RWybSbEIXr56RyulkBlFKzSWVKL/jdM4UYiox8hChQWusnlVKTgDeUUlFSCWI/o7U+luemCZExSSwqhBAi52TaTYj/214d36N4rQAAAChJREFUCwAAAAAM8reexa6SCNjJB4CdfADYyQeAnXwA2MkHgJ18ANgFPOhdODrYcM0AAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "sns.scatterplot(data_pca['PC1'],data_pca['PC2'],hue=data.sum())\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### GSE148729"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Gene name conversion from ENSEMBL ID to HGNC symbol"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "data=pd.read_csv('../data/gse148729/GSE148729_Calu3_totalRNA_readcounts.tsv',low_memory=False,index_col=0, sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CompletedProcess(args=['/usr/local/bin/Rscript', '--verbose', '../code/gse148729_gene_name_conversion.R'], returncode=0, stdout=b'', stderr=b\"running\\n  '/Library/Frameworks/R.framework/Versions/4.0/Resources/bin/R --no-echo --no-restore --file=../code/gse148729_gene_name_conversion.R'\\n\\n\")\n",
      "Error: \n",
      "b\"running\\n  '/Library/Frameworks/R.framework/Versions/4.0/Resources/bin/R --no-echo --no-restore --file=../code/gse148729_gene_name_conversion.R'\\n\\n\"\n",
      "Result: \n",
      "b''\n"
     ]
    }
   ],
   "source": [
    "runR(\"../code/gse148729_gene_name_conversion.R\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "name_conversion = pd.read_csv('../data/gse148729/gse148729_name_conversion.csv', sep=',', index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "name_conversion = name_conversion[name_conversion.hgnc_symbol.notna()].drop_duplicates('hgnc_symbol')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "data = data.reset_index()\n",
    "data['ensembl_gene_id']= data.gene_id.str.split('.',expand=True)[0]\n",
    "data = data.merge(name_conversion, \n",
    "                  left_on='ensembl_gene_id', \n",
    "                  right_on='ensembl_gene_id').set_index('hgnc_symbol').drop(columns=['gene_id', 'ensembl_gene_id'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Metadata and gene expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "series_matrix=pd.read_csv('../data/gse148729/GSE148729-GPL20301_series_matrix.txt',sep='\\t',skiprows=66,low_memory=False,index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata = create_metadata_structure(series_matrix)\n",
    "metadata = add_properties_to_metadata(metadata, sep='-', properties={'Treatment':1, 'Time':2, 'Replicate':3} )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "# change sample ID to Sample_geo_accession\n",
    "d = dict(map(reversed, metadata['ID'].to_dict().items()))\n",
    "data.columns = data.columns.to_series().map(d)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata.to_csv('../data/gse148729/meta.csv')\n",
    "data.to_csv('../data/gse148729/GSE148729_raw.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###### Differencial expression analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CompletedProcess(args=['/usr/local/bin/Rscript', '--verbose', '../code/gse148729_DE.R'], returncode=0, stdout=b'[1] \"vst done\"\\n                                  ID Treatment Time Replicate\\nGSM4477956   Calu3_totalRNA-S1-24h-A        S1  24H         A\\nGSM4477957   Calu3_totalRNA-S1-24h-B        S1  24H         B\\nGSM4477950 Calu3_totalRNA-mock-24h-A      mock  24H         A\\nGSM4477951 Calu3_totalRNA-mock-24h-B      mock  24H         B\\n                                  ID Treatment Time Replicate\\nGSM4477962   Calu3_totalRNA-S2-24h-A        S2  24H         A\\nGSM4477963   Calu3_totalRNA-S2-24h-B        S2  24H         B\\nGSM4477950 Calu3_totalRNA-mock-24h-A      mock  24H         A\\nGSM4477951 Calu3_totalRNA-mock-24h-B      mock  24H         B\\n                                 ID Treatment Time Replicate\\nGSM4477958   Calu3_totalRNA-S1-4h-A        S1   4H         A\\nGSM4477959   Calu3_totalRNA-S1-4h-B        S1   4H         B\\nGSM4477952 Calu3_totalRNA-mock-4h-A      mock   4H         A\\nGSM4477953 Calu3_totalRNA-mock-4h-B      mock   4H         B\\n                                 ID Treatment Time Replicate\\nGSM4477964   Calu3_totalRNA-S2-4h-A        S2   4H         A\\nGSM4477965   Calu3_totalRNA-S2-4h-B        S2   4H         B\\nGSM4477952 Calu3_totalRNA-mock-4h-A      mock   4H         A\\nGSM4477953 Calu3_totalRNA-mock-4h-B      mock   4H         B\\n[1] \"de done\"\\n', stderr=b'running\\n  \\'/Library/Frameworks/R.framework/Versions/4.0/Resources/bin/R --no-echo --no-restore --file=../code/gse148729_DE.R\\'\\n\\nLoading required package: S4Vectors\\nLoading required package: stats4\\nLoading required package: BiocGenerics\\nLoading required package: parallel\\n\\nAttaching package: \\xe2\\x80\\x98BiocGenerics\\xe2\\x80\\x99\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:parallel\\xe2\\x80\\x99:\\n\\n    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\\n    clusterExport, clusterMap, parApply, parCapply, parLapply,\\n    parLapplyLB, parRapply, parSapply, parSapplyLB\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:stats\\xe2\\x80\\x99:\\n\\n    IQR, mad, sd, var, xtabs\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:base\\xe2\\x80\\x99:\\n\\n    Filter, Find, Map, Position, Reduce, anyDuplicated, append,\\n    as.data.frame, basename, cbind, colnames, dirname, do.call,\\n    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,\\n    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,\\n    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,\\n    tapply, union, unique, unsplit, which, which.max, which.min\\n\\n\\nAttaching package: \\xe2\\x80\\x98S4Vectors\\xe2\\x80\\x99\\n\\nThe following object is masked from \\xe2\\x80\\x98package:base\\xe2\\x80\\x99:\\n\\n    expand.grid\\n\\nLoading required package: IRanges\\nLoading required package: GenomicRanges\\nLoading required package: GenomeInfoDb\\nLoading required package: SummarizedExperiment\\nLoading required package: Biobase\\nWelcome to Bioconductor\\n\\n    Vignettes contain introductory material; view with\\n    \\'browseVignettes()\\'. To cite Bioconductor, see\\n    \\'citation(\"Biobase\")\\', and for packages \\'citation(\"pkgname\")\\'.\\n\\nLoading required package: DelayedArray\\nLoading required package: matrixStats\\n\\nAttaching package: \\xe2\\x80\\x98matrixStats\\xe2\\x80\\x99\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:Biobase\\xe2\\x80\\x99:\\n\\n    anyMissing, rowMedians\\n\\n\\nAttaching package: \\xe2\\x80\\x98DelayedArray\\xe2\\x80\\x99\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:matrixStats\\xe2\\x80\\x99:\\n\\n    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:base\\xe2\\x80\\x99:\\n\\n    aperm, apply, rowsum\\n\\n\\xe2\\x94\\x80\\xe2\\x94\\x80 Attaching packages \\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80 tidyverse 1.3.0 \\xe2\\x94\\x80\\xe2\\x94\\x80\\n\\xe2\\x9c\\x94 ggplot2 3.3.2     \\xe2\\x9c\\x94 purrr   0.3.4\\n\\xe2\\x9c\\x94 tibble  3.0.4     \\xe2\\x9c\\x94 dplyr   1.0.2\\n\\xe2\\x9c\\x94 tidyr   1.1.2     \\xe2\\x9c\\x94 stringr 1.4.0\\n\\xe2\\x9c\\x94 readr   1.4.0     \\xe2\\x9c\\x94 forcats 0.5.0\\n\\xe2\\x94\\x80\\xe2\\x94\\x80 Conflicts \\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80 tidyverse_conflicts() \\xe2\\x94\\x80\\xe2\\x94\\x80\\n\\xe2\\x9c\\x96 ggplot2::Position() masks BiocGenerics::Position(), base::Position()\\n\\xe2\\x9c\\x96 dplyr::collapse()   masks IRanges::collapse()\\n\\xe2\\x9c\\x96 dplyr::combine()    masks Biobase::combine(), BiocGenerics::combine()\\n\\xe2\\x9c\\x96 dplyr::count()      masks matrixStats::count()\\n\\xe2\\x9c\\x96 dplyr::desc()       masks IRanges::desc()\\n\\xe2\\x9c\\x96 tidyr::expand()     masks S4Vectors::expand()\\n\\xe2\\x9c\\x96 dplyr::filter()     masks stats::filter()\\n\\xe2\\x9c\\x96 dplyr::first()      masks S4Vectors::first()\\n\\xe2\\x9c\\x96 dplyr::lag()        masks stats::lag()\\n\\xe2\\x9c\\x96 purrr::reduce()     masks GenomicRanges::reduce(), IRanges::reduce()\\n\\xe2\\x9c\\x96 dplyr::rename()     masks S4Vectors::rename()\\n\\xe2\\x9c\\x96 purrr::simplify()   masks DelayedArray::simplify()\\n\\xe2\\x9c\\x96 dplyr::slice()      masks IRanges::slice()\\nestimating size factors\\nestimating dispersions\\ngene-wise dispersion estimates: 10 workers\\nmean-dispersion relationship\\nfinal dispersion estimates, fitting model and testing: 10 workers\\nestimating size factors\\nestimating dispersions\\ngene-wise dispersion estimates: 10 workers\\nmean-dispersion relationship\\nfinal dispersion estimates, fitting model and testing: 10 workers\\nestimating size factors\\nestimating dispersions\\ngene-wise dispersion estimates: 10 workers\\nmean-dispersion relationship\\nfinal dispersion estimates, fitting model and testing: 10 workers\\nestimating size factors\\nestimating dispersions\\ngene-wise dispersion estimates: 10 workers\\nmean-dispersion relationship\\nfinal dispersion estimates, fitting model and testing: 10 workers\\nestimating size factors\\nestimating dispersions\\ngene-wise dispersion estimates: 10 workers\\nmean-dispersion relationship\\nfinal dispersion estimates, fitting model and testing: 10 workers\\n')\n",
      "Error: \n",
      "b'running\\n  \\'/Library/Frameworks/R.framework/Versions/4.0/Resources/bin/R --no-echo --no-restore --file=../code/gse148729_DE.R\\'\\n\\nLoading required package: S4Vectors\\nLoading required package: stats4\\nLoading required package: BiocGenerics\\nLoading required package: parallel\\n\\nAttaching package: \\xe2\\x80\\x98BiocGenerics\\xe2\\x80\\x99\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:parallel\\xe2\\x80\\x99:\\n\\n    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\\n    clusterExport, clusterMap, parApply, parCapply, parLapply,\\n    parLapplyLB, parRapply, parSapply, parSapplyLB\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:stats\\xe2\\x80\\x99:\\n\\n    IQR, mad, sd, var, xtabs\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:base\\xe2\\x80\\x99:\\n\\n    Filter, Find, Map, Position, Reduce, anyDuplicated, append,\\n    as.data.frame, basename, cbind, colnames, dirname, do.call,\\n    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,\\n    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,\\n    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,\\n    tapply, union, unique, unsplit, which, which.max, which.min\\n\\n\\nAttaching package: \\xe2\\x80\\x98S4Vectors\\xe2\\x80\\x99\\n\\nThe following object is masked from \\xe2\\x80\\x98package:base\\xe2\\x80\\x99:\\n\\n    expand.grid\\n\\nLoading required package: IRanges\\nLoading required package: GenomicRanges\\nLoading required package: GenomeInfoDb\\nLoading required package: SummarizedExperiment\\nLoading required package: Biobase\\nWelcome to Bioconductor\\n\\n    Vignettes contain introductory material; view with\\n    \\'browseVignettes()\\'. To cite Bioconductor, see\\n    \\'citation(\"Biobase\")\\', and for packages \\'citation(\"pkgname\")\\'.\\n\\nLoading required package: DelayedArray\\nLoading required package: matrixStats\\n\\nAttaching package: \\xe2\\x80\\x98matrixStats\\xe2\\x80\\x99\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:Biobase\\xe2\\x80\\x99:\\n\\n    anyMissing, rowMedians\\n\\n\\nAttaching package: \\xe2\\x80\\x98DelayedArray\\xe2\\x80\\x99\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:matrixStats\\xe2\\x80\\x99:\\n\\n    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges\\n\\nThe following objects are masked from \\xe2\\x80\\x98package:base\\xe2\\x80\\x99:\\n\\n    aperm, apply, rowsum\\n\\n\\xe2\\x94\\x80\\xe2\\x94\\x80 Attaching packages \\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80 tidyverse 1.3.0 \\xe2\\x94\\x80\\xe2\\x94\\x80\\n\\xe2\\x9c\\x94 ggplot2 3.3.2     \\xe2\\x9c\\x94 purrr   0.3.4\\n\\xe2\\x9c\\x94 tibble  3.0.4     \\xe2\\x9c\\x94 dplyr   1.0.2\\n\\xe2\\x9c\\x94 tidyr   1.1.2     \\xe2\\x9c\\x94 stringr 1.4.0\\n\\xe2\\x9c\\x94 readr   1.4.0     \\xe2\\x9c\\x94 forcats 0.5.0\\n\\xe2\\x94\\x80\\xe2\\x94\\x80 Conflicts \\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80\\xe2\\x94\\x80 tidyverse_conflicts() \\xe2\\x94\\x80\\xe2\\x94\\x80\\n\\xe2\\x9c\\x96 ggplot2::Position() masks BiocGenerics::Position(), base::Position()\\n\\xe2\\x9c\\x96 dplyr::collapse()   masks IRanges::collapse()\\n\\xe2\\x9c\\x96 dplyr::combine()    masks Biobase::combine(), BiocGenerics::combine()\\n\\xe2\\x9c\\x96 dplyr::count()      masks matrixStats::count()\\n\\xe2\\x9c\\x96 dplyr::desc()       masks IRanges::desc()\\n\\xe2\\x9c\\x96 tidyr::expand()     masks S4Vectors::expand()\\n\\xe2\\x9c\\x96 dplyr::filter()     masks stats::filter()\\n\\xe2\\x9c\\x96 dplyr::first()      masks S4Vectors::first()\\n\\xe2\\x9c\\x96 dplyr::lag()        masks stats::lag()\\n\\xe2\\x9c\\x96 purrr::reduce()     masks GenomicRanges::reduce(), IRanges::reduce()\\n\\xe2\\x9c\\x96 dplyr::rename()     masks S4Vectors::rename()\\n\\xe2\\x9c\\x96 purrr::simplify()   masks DelayedArray::simplify()\\n\\xe2\\x9c\\x96 dplyr::slice()      masks IRanges::slice()\\nestimating size factors\\nestimating dispersions\\ngene-wise dispersion estimates: 10 workers\\nmean-dispersion relationship\\nfinal dispersion estimates, fitting model and testing: 10 workers\\nestimating size factors\\nestimating dispersions\\ngene-wise dispersion estimates: 10 workers\\nmean-dispersion relationship\\nfinal dispersion estimates, fitting model and testing: 10 workers\\nestimating size factors\\nestimating dispersions\\ngene-wise dispersion estimates: 10 workers\\nmean-dispersion relationship\\nfinal dispersion estimates, fitting model and testing: 10 workers\\nestimating size factors\\nestimating dispersions\\ngene-wise dispersion estimates: 10 workers\\nmean-dispersion relationship\\nfinal dispersion estimates, fitting model and testing: 10 workers\\nestimating size factors\\nestimating dispersions\\ngene-wise dispersion estimates: 10 workers\\nmean-dispersion relationship\\nfinal dispersion estimates, fitting model and testing: 10 workers\\n'\n",
      "Result: \n",
      "b'[1] \"vst done\"\\n                                  ID Treatment Time Replicate\\nGSM4477956   Calu3_totalRNA-S1-24h-A        S1  24H         A\\nGSM4477957   Calu3_totalRNA-S1-24h-B        S1  24H         B\\nGSM4477950 Calu3_totalRNA-mock-24h-A      mock  24H         A\\nGSM4477951 Calu3_totalRNA-mock-24h-B      mock  24H         B\\n                                  ID Treatment Time Replicate\\nGSM4477962   Calu3_totalRNA-S2-24h-A        S2  24H         A\\nGSM4477963   Calu3_totalRNA-S2-24h-B        S2  24H         B\\nGSM4477950 Calu3_totalRNA-mock-24h-A      mock  24H         A\\nGSM4477951 Calu3_totalRNA-mock-24h-B      mock  24H         B\\n                                 ID Treatment Time Replicate\\nGSM4477958   Calu3_totalRNA-S1-4h-A        S1   4H         A\\nGSM4477959   Calu3_totalRNA-S1-4h-B        S1   4H         B\\nGSM4477952 Calu3_totalRNA-mock-4h-A      mock   4H         A\\nGSM4477953 Calu3_totalRNA-mock-4h-B      mock   4H         B\\n                                 ID Treatment Time Replicate\\nGSM4477964   Calu3_totalRNA-S2-4h-A        S2   4H         A\\nGSM4477965   Calu3_totalRNA-S2-4h-B        S2   4H         B\\nGSM4477952 Calu3_totalRNA-mock-4h-A      mock   4H         A\\nGSM4477953 Calu3_totalRNA-mock-4h-B      mock   4H         B\\n[1] \"de done\"\\n'\n"
     ]
    }
   ],
   "source": [
    "### run deseq2 in R\n",
    "runR('../code/gse148729_DE.R')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### PC analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "vst = pd.read_csv('../results/gse148729/gse148729_vst.csv', sep=',', index_col=0)\n",
    "model=PCA(2)\n",
    "vst_pca=model.fit_transform(vst.T)\n",
    "vst_pca=pd.DataFrame(vst_pca,index=vst.columns,columns=['PC1','PC2'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAdkAAAEECAYAAABk507zAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjMsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+AADFEAAAgAElEQVR4nO3deXxTVd4/8E/27nSjLIWW/YgIFYrCKPMA4mjlJ6MsP5fBcWAQBBEBRQWko75EcENEHPQHBR1AtodFXLDq8AAP6qgMCoqEI5R9KXQBuqRJmia/P9IwKTRNt5vbNJ/36+WruTfp7ZczDJ+ec889R+NyuUBERESNT6t2AURERM0VQ5aIiEghDFkiIiKFMGSJiIgUwpAlIiJSiF7tAupq3759LpPJpHYZ9Waz2RDM9SuN7eMb26ZmbB/fbDYbKioq8tPT01uqXUuoCbqQNZlM6N69u9pl1JvZbA7q+pXG9vGNbVMzto9vZrMZFovlhNp1hCIOFxMRESmEIUtERKQQhiwREZFCgu6eLBERNY69e/cm6fX6LAA3gJ2u+nACOOBwOB5JT0+/UN0HAhqyQggdgGUABIAKAGMBaAB8AMAF4ACAyVJKZyDrIiIKRXq9Pqt169bdW7ZseVGr1XIh+zpyOp2avLy863Nzc7MA/LG6zwT6N5dhACClvBXA3wC8WfnfHCnl7+EO3HsCXFOTUWIvQVl52TWviYgUckPLli2LGLD1o9VqXS1btrwM90hAtQLak5VSfiSE+LTyMBXAeQD/B8CuynOfA7gDwBZf17DZbDCbzYrWqSSr1XpN/TqdDi3atMC4f47D0+lPo2tsV0zYPgHP3fQc2rrawlJiUanawKuufciNbVMzto9vVqvV11taBmzDVLafzw5rwO/JSikdQoh/ABgOYBSAu6WUnv+RiwG0qOn7m+tzspdslzCw3UA8sfMJRBmiEBcWh05xndAqspUKVaqHzzr6xrapGdvHt8rnZNUuIySpMvFJSvkXIcSzAL4HEO71VjSAS2rUpJaL1osAgLiwOIzpMQarDq5Ckb0Ik2+cjBamGn/fICIKanPmzGl36NChyMLCQr3NZtO2adPGrtfrnTfeeGPJzJkzz6ldX2MI9MSnPwNoJ6WcD8AC98ysfwshBkkpdwK4C8COQNakpsu2y3jl+1dg1BkxNX0qHs5+GCkxKbgh4Qa8tuc1dIntgt5JvWHQGdQulYio0c2dO/c0AKxcuTLh6NGjYS+88MIZtWtqbIHuyW4G8L4Q4n8BGABMA2AGsEwIYax8vTHANakmQh+BP/f4M8Z9MQ63JN+CP3b6I4Z1HoYoQxS6xXVDakwqA5aIQsrOnTuj16xZ03Lp0qVHBw0adEOPHj1KTp8+Hda3b9/i4uJindlsjmzfvr11yZIlx06ePGmYNWtWB7vdrjEaja758+cfT0lJKVf7z+At0BOfSgHcV81bAwNZR1Nh0BnQPro9UmJSMGv3LMzpPwctjC0QbYrGA9c9gAhDhNolEhGp5vz586Z//OMfv7Vp06b8pptuunHNmjXm7t27WwcOHNjz4sWLurlz57Z/6KGHzt91111F27dvj54/f367d99995jadXvjYhQqumy7jHnfzcOJohN4vPfjePWHV7Hvwj48fdPTvB9LRCEvOjrakZqaageAsLAwZ48ePawAEBkZWWG1WjVHjx4NX7ZsWZvly5e3cblcMBgMTW6NBYasijzDxWNuGINOsZ3Qv01/6DQ6ROjZgyUi0mg0Nb6fkpJiHTduXO6tt95aajabw7755puoAJVWawxZFRl0Bog4Ue1rIiKq2XPPPXdqzpw5qW+++abWbrdrZ82adVLtmq7GkFWZd6AyXIkoFD388MMFnteDBg0qHjRoUDEAfP/99/s9571fZ2dnH/S8Xrt27eFA1VkfXBCaiIhIIQxZIiIihTBkiYiIFMKQJSIiUghDloiISCEMWSIiIoUwZImISFXnz5/X33rrrb3MZnPYlClTOmRnZ8d4v9+vX780tWprKD4nS0REtbL6uxPxb28/nJxXbDO2jDbZnxjS9cxD/VMLG3JNu92umTlzZqrRaGxySyI2BvZkiYjIr9XfnYh/6dODqReKbUYXgAvFNuNLnx5MXf3difiGXPf5559v98ADD+QlJCQ0qd1zGgt7skRE5Nfb2w8n2xzOKh0zm8OpfXv74eT69mZXrlyZEBcX57jzzjuLli1b1sZzfuHChe2ysrIqPMclJSW6+leuLoYsERH5lVdsM9blfG1s3bo1EQBGjRoVk5OTE/7MM890jI6OdkyfPv10RkZGkedzvCdLRETNWstok/1CNYHaMtpkr+81N23aJD2vR40aJV566aUTS5YsaV3f6zVFvCdLRER+PTGk6xmTXltlcpJJr3U+MaTrGbVqCgbsyRIRkV+e+66NPbvYY+PGjRIAFi9efPzq97x34Ak2DFkiIqqVh/qnFjZWqIYKDhcTEREphCFLRESkEIYsERGRQhiyRERECmHIEhERKYQhS0REqvruu+8iR40aJQDgp59+Ch8xYoQYNWqUePDBB7vm5ubqgWtXfcrOzo6ZMmVKBxXKrROGLBER1c6e5fF4o1tPvBCbjje69cSe5Q3aHAAAFi5c2Opvf/tbqt1u1wDA3LlzUzIzM09u3LhRDhky5OLixYuDegUohiwREfm3Z3k8vpiVipLzRsAFlJw34otZqQ0N2tTUVNs777yT4zlevHjx0d69e5cBQEVFhcZkMrkaWrqauBgFERH5t+vVZDhsVTtmDpsWu15Nxk3j6r1AxYgRIy4dO3bsyprIbdu2LQeAb775JnLDhg1Ja9eulYB7Jx7PkDIAFBcX67p162ap788NFIYsERH5V3Kh+t12fJ1vgA0bNsQtX768zdKlSw8nJSU5ACAqKqrCs/Qi4L4n+9lnnzV4uFppDFkiIvIvKsnuHiqu5nwjWrNmTfzGjRtbrlmzRiYkJFT4/46mjfdkiYjIv4HPnoHeVGUXHuhNTgx8ttF24XE4HFiwYEGKxWLRPfroo11GjRol5s2b17axrq8G9mSJiMg/z33XXa8mo+SCEVFJdgx89kxD7sd6dOzY0f7xxx8fAoC9e/fuq+4zV+/Ek5GRUeS9sXtTxZAlIqLauWlcYWOEaijhcDEREZFCGLJEREQKCdhwsRDCAGAFgA4ATADmAjgI4AMALgAHAEyWUjp9XIKIiCioBLIn+xCAAinl7wHcBeAdAG8CmFN5TgPgngDWQ0REpKhATnz6bwAbvY4dANIB7Ko8/hzAHQC21HQRm80Gs9msSIGBYLVag7p+pbF9fGPb1Izt45vValW7hJAVsJCVUpYAgBAiGu6wnQPgDSmlZ13KYgAt/F3HZDKhe/fuitWpNLPZHNT1K43t4xvbpmZsH9/MZjMslqa3AqHdbtdMmzatQ25urrG8vFw7YcKEs8OGDbsMAOvWrYtfu3Zt0tatWw8B7l14vB/j8az4tHjx4uMqlV8rAZ34JIRoD2AHgFVSyjUAvO+/RgO4FMh6iIio9tbL9fGDNwzu2esfvdIHbxjcc71c36BlDdeuXRsfGxvr2Lx5s3z//fd/e+WVV1IA93Z3mzdvTnS5gnpvAAABDFkhRCsAXwJ4Vkq5ovL0T0KIQZWv7wKwO1D1EBFR7a2X6+Nf2/Naan5ZvtEFF/LL8o2v7XkttSFBO2LEiIszZ868smKUTqdDfn6+7vXXX283Z86cU41TuboCeU92NoA4AJlCiMzKc1MBvC2EMAIwo+o9WyIiaiLe2/9esr3CXqVjZq+wa9/b/17y/eL+ei1QER0d7QSAoqIi7aRJkzpPmjTpzIwZMzrMmTPnVHh4eJUnTbgLjx9Syqlwh+rVBgaqBiIiqp+CsoJqd9vxdb62Tp48aZg0aVKX++6770KnTp2sZ86cCfNs4n7y5Mnw2bNnt583b94p7sJDRETNVkJ4gj2/LP+aQE0IT6j3Ljy5ubn6sWPHdps9e/bJIUOGFAPAV1999SsAHDt2zDh16tRO8+bNC+phY674REREfk1Mm3jGqDNWGcI16ozOiWkT670Lz6JFi9qUlJTo33333bajRo0So0aNEhaLRdPwapsO9mSJiMgvz33X9/a/l1xQVmBMCE+wT0ybeKa+92MBYP78+acAVNtT9d6ZB+AuPERE1MzdL+4vbEiohiIOFxMRESmEIUtERKQQhiwREZFCGLJEREQKYcgSEREphLOLiYhIFXPmzGl36NChyMLCQr3NZtO2adPGHhsbW56VlXW0Ptf7+eefwy9evKgbOHBgSWPWWVZWplm3bl3C2LFj8+v6vQxZIiJSxdy5c08DwMqVKxOOHj0a9sILL9R7YQsA+Oyzz+ISExPLGztkc3NzDVu2bElkyBIRUVDbuXNn9IIFC5INBoNr5MiRecnJyfZFixYla7VaJCcn2954440TZWVlmhkzZnQoKSnRXb58WX/vvffmDx069NK2bdsS9Hq9q1evXqWZmZkd0tLSSo4ePRqekpJijY+PL9+3b1+0wWBwrly58ojFYtE+9dRTqUVFRXoAyMzMPNWrV6+yQYMG3dCzZ8+SkydPhsXFxZVnZWXlvP32221OnjwZ/sorr7SZOXPmubr8eRiyRETUpNjtdu0nn3xy0Ol0YsiQITesW7fuUKtWrRxz585tu3r16oS0tDTL0KFDC4cPH37pzJkzhtGjR4sJEybkDR06tCAxMbG8X79+lrKyMt0999xTMGDAgNLbbrutx4wZM05nZmaeHTlypPj111/DPvroo/h+/foVT5gwIe+3334zPfvssx22bNkiz58/b1q5cqVMSUkpv/fee6/bs2dP5BNPPHEuJycnvK4BCzBkiYioiUlJSbECQF5enr6wsNAwefLkzgBgs9k0/fr1K7rzzjsvr1ixIumrr76Ki4yMrKioqKh2veM+ffpYACAqKqqie/fuZQAQHR3tsFqt2pycnPC9e/fGfPnll/EAUFJSove8n5KSUg4ASUlJdpvN1qAJwgxZIiJqUjQad2YmJiY6EhMT7VlZWUdiY2MrPvnkkxaRkZHOJUuWtEpLSyudMGFC3o4dO6L/9a9/taj8PpfL5brmOtVJTU213n333YUPPPBA4fnz5/WrVq1K9PU9Wq0WLperXhsXMGSJqFmIbROLMyVn4HQ54f0PbZg+DOXOckQaIhFrilWxQqornU6HZ5999tTYsWO7uFwuTURERMXChQuPaTQavPTSSymff/55fExMjEOn07msVqumZ8+eloULF7br2rWr1d+1p0+ffm7GjBkdNm7c2NJisWgnTZp01tdnk5KSysvLyzXPP/988osvvlinyVka77+MwcBsNru6d++udhn1ZjabEcz1K43t4xvbpmbHC49j2CfDrjm/4s4V+OsXf0X2yGwkRyWrUJn6zGYzLBbL3vT09L7e5/fv3388LS2tzjNmqar9+/cnpqWldajuPS5GQUREpBCGLBERkUIYskRERAphyBIRESmEs4uJqFkwaU3IHpld7ezi7JHZiDREqlgdhSqGLBE1C5fOXeLsa2pyOFxMRERBYfny5Yl2u71ei0KohSFLRES1smnTpviMjIyeffv2Tc/IyOi5adOm+ED+/A8++KBNRUVFIH9kg3G4mIiI/Nq0aVP8ggULUu12uxYA8vPzjQsWLEgFgJEjRxbW97re29yVlZVp7rjjjhtatWpl79atmyUnJyfcYrHoFi9enLNz586YixcvGiZNmtRpzJgxF7x36hk9enS9f77S2JMlIiK/li1bluwJWA+73a5dtmyZIsto9erVq3T9+vW/3XzzzUWbNm2KHzNmTH5cXFz5u+++e9Tzszdv3iybcsACDFkiIqqF/Px8Y13O14f3rPBevXpZAKB169b2q8Md+M9OPU0dQ5aIiPxKTEy01+V8bZlMJmd+fr4BAH788ccIz/nqdsPRaDRwOp0aX+83RQxZIiLya/z48WeMRqPT+5zRaHSOHz++TrvSXO2OO+4oOnfunGn48OHis88+i4+IiPA5s6lXr17Ff/nLX7oG08Y2nPhERER+eSY3LVu2LDk/P9+YmJhoHz9+/JmGTHoCgLi4uIpNmzZJX++PHz8+z/P6nXfeOe55PXjw4OKG/NxAYcgSEVGtjBw5srChoRpqOFxMRESkEIYsERGRQhiyRERECgn4PVkhRD8Ar0opBwkhugD4AIALwAEAk6WUzpq+n4iIKFgEtCcrhHgGQBaAsMpTbwKYI6X8PQANgHsCWQ8REZGSAt2TzQEwAsCqyuN0ALsqX38O4A4AW2q6gM1mg9lsVqxApVmt1qCuX2lsH9/YNjVj+/hmtQbF4kgN4r0Gstq1eAtoyEopNwkhOnid0kgpPU8VFwNo4e8aJpMpqPeMNJvNQV2/0tg+vrFtasb28c1sNsNisTToGna7XfPYY491AYC33norZ9q0aZ0BYMmSJUeMRmPwrA4RYGo/J+t9/zUawCW1CiEiIt8ee+yxLgcPHowCgKFDh/ZyOBwaz/msrKzD9b3uypUrE3bt2hVrs9k0BQUFxgcffPD8jh07Yo8dOxb+5JNPni4tLdV++OGHrQwGg7Ndu3a2BQsWnHA4HJg2bVrH8+fPGx0Oh+a555476bnehQsX9BMmTOgyZcqUM0OGDFF9wQq1Q/YnIcQgKeVOAHcB2KFyPUREVAO73a61293LFV+9zGJ9WSwW7dq1aw+vX78+bvXq1a22bt16aNeuXdHvv/9+q5MnT4Z9/PHHB2NiYpyzZ89uv2LFisTy8nJN27ZtbUuXLj36yy+/hO3atSsmJiamoqCgwDB+/Pgus2bNOtW/f//SxqitodR+hOcpAC8KIf4FwAhgo8r1EBFRNd56660cvV5fZVhYr9e7Fi1alNPQa3fr1s0CAC1atKjo0KGDVavVIjY21mG1WrWpqallMTExTgC4+eabi48cORJ+/PjxsD59+pQCQM+ePa2PP/74BQDYs2dPTHl5ucbpbDoPqQQ8ZKWUx6WU/Stf/yalHCil/J2U8q9SyuDa8p6IKERMmzats2eI2MPhcGimTp3auaHX9rWjjkajwYkTJ8JLSkq0APDDDz9Ep6amWjt16mTdv39/BADk5OQYJ06c2BEAMjIyCl5//fVjzz//fAfP96hN7eFiIiIKIkaj0anX611XB64SdDqda+LEiWcffPBBodFoXO3atbNlZmaeBoDp06d3GDlypHA6nZg9e/Yps9kcDgA9evSwZmRkFGRmZrZfuHDhCaVr9EcTTFsGAYDZbHYF8wxCzoCsGdvHN7ZNzdg+vlXOLt6bnp7e1/v8/v37j6elpeXX5hqcXezb/v37E9PS0jpU9x57skRE5JfRaHR5zyJuyIziUFLjmLUQoqsQYpMQYrUQoqvX+XeVL42IiCi4+evJLgUwH4ABwEdCiIeklD8BuE7xyogshYCtmsfcTNFARHzg6yEiqiO/w8VSyi8BQAhxBMBmIUQG3Av6EynLVgws6nXt+ak/M2SJKCj4m+LsEEIME0LopJQSwOMAPgXQWvnSiIiIgpu/kP0rgJGoXFNYSrkDwDQAdoXrIiIiCno1DhdLKU8JIZ6UUhYCgBAiCcAvUsobA1IdUW3w3i1R0FqwYEHrH374IUar1boA4KmnnjrTt29fCwDMnj27fceOHa3jx4/PU7fK+qsxZIUQAwGsFELcKKW8CCANQJYQYrSU8uuAVEjkD+/dEiluwIABva1W6zWjn2FhYc6vv/76p/pc89dffw3bvXt37ObNmw9ptVr89NNP4TNnzuy4atWq36ZPn97x9OnTpo4dOwb1Pn3+Jj7NBTCwMmAhpfxKCPEHAMsB/F7p4ijEmaLdQVndeSIKqOoCtqbztZGUlOS4cOGCceXKlYm333775d69e5dt3brVfPbsWcOUKVPO7tixw+/2p02d34lPUsrj3ieklL+h6hZ1RMqIiAfiUq/9j71TomahZcuWjsWLFx/58ccfo+6///7rbrvtth7btm1r0alTJ3tT2UWnofz1ZLVCCK2U8kqoCiF0cO+YQ0REVG+HDx82xcTEVLz99tvHAWDPnj0RkydP7jpw4MDihISEZrFhjL+e7IcA1goh0oQQ0UKI6wGsBrBe+dKIiKg5O3DgQHhmZmZKWVmZBgC6detmjYiIqNDpdGqX1mj8zS5eKoS4DOBNAG0BnADwvpSSIUtNB+/dEgWl4cOHXzpy5EjYvffe2z08PNzpcrkwffr007Gxsc2iFwv4n108GcAMABUAHpdSZgekKqK6iIjnfVoihYWFhTl9zS5uyHWffvrp3Keffjq3uvdmzZp1tiHXbgr83ZMdDUAAiAGwCgBDlogoBNX3MZ1Q5++erFVKaZdS5oOTnYiIiOqkLs83aRSrgoiIqBnyN1zcQwixBu6A9bwGAEgp/6RoZUREREHOX8je5/X6PSULISIiam78PcKzK1CFEBERNTd+N20nIiJSSnW78Oh0OtdLL72UotVqYTAYnAsXLjzWunVrh9q11gdDloiIas2zG09Ddt/x8LULT1RUVEVmZubJ3r17l2VlZSUuXry49csvv3y6sf4MgcSQJSKiWvMsSNGQ3Xc8fO3CU1hYqG/btm05AFRUVGhMJpOroT9LLQ1uJCIiav4GDBjQu2/fvularTs2tFot+vbtmz5gwIDe9b2mr114PAH7zTffRG7YsCFp4sSJ5xvpjxFw7MkSEZFfnp6r0+leRdHztSE92pp24dm+fXvM8uXL2yxduvRwUlJSUN6PBdiTJSKiWvCsUezdk/U+Xx++duHZtm1b7Lp165LWrFkjO3fubG949ephT5aIiPzyTHLq27dvOuDuyf773//e25BrVrcLzxNPPHH65Zdf7tCyZUv7o48+2gUA+vTpUzx79uyg3CyAIUtERLXm2Y2nobvveFS3C8+IESP2Nca1mwKGLBER1Rp346kb3pMlIiJSCEOWiCh0OZ1OJ3dYa4DK9vM5dM6QJSIKXQfy8vJaMGjrx+l0avLy8loAOODrM7wnS0QUohwOxyO5ublZubm5N4CdrvpwAjjgcDge8fUB1UNWCKEFsARAGgAbgEeklEfUrYqIqPlLT0+/AOCPatfRnDWF31zuBRAmpfwdgJkAFqhcDxERUaNQvScLYACAbACQUn4nhOhb04dtNhvMZnNAClOC1WoN6vqVxvbxjW1TM7aPb1arVe0SQlZTCNkYAJe9jiuEEHopZbVrVZpMJnTv3j0wlSnAbDYHdf1KY/v4xrapGdvHN7PZDIvFonYZIakpDBcXAYj2Otb6ClgiIqJg0hRC9hsAQwFACNEfwC/qlkNERNQ4msJw8RYAfxBCfAtAA2CsyvUQERE1CtVDVkrpBDBR7TqIiIgaW1MYLiYiImqWGLJEREQKYcgSEREphCFLRESkEIYsERGRQhiyRERECmHIEhERKYQhS0REpBDVF6NoCi6X2XHushXfHS3Eje1bIDU+EnGRRrXLIiKiIBfyIWt3OPHxvrPI3PrrlXOPDOiIqbd3RXSYQcXKiIgo2IX8cPElix2vZcsq597/9jhKbNwIiIiIGibkQ9YFoKy8osq5CqcLTqc69RARUfMR8iEbadJhVHq7Kud+1zkB4caQbxoiImqgkL8nG2Uy4JkMgR5tY/BP8wXc3DEe9/dtj/hIk9qlERFRkAv5kAWA+EgTRvdLxfA+yQjX66DTsRdLREQNx5CtpNVqEGXibGIiImo8DFki8s9SCNiKAVM0EBGvdjVEQSNkx0VLrA44KpzXvCaiatiKgUW93F+JqNZCMmTPXCrDPX//GkfySnCy0IJ7/v41juWXql0WUdNjKQQungBKL7iPSy+4jy2F6tZFFCRCcrg4TK9F+7gIDP/7tzDqtWgXF47YCN6PJbqGpwfrkXW7++vUnzlsTFQLIdGTLSy145LFfuW1TqtB5t3Xo6y8ApfLyjHzruu4VjFRdUzR7kB95J/u40f+6T42RatbF1GQaPY92YsWO1793IzEaBMeG9QF87eZ0aZFGNrGhkO0ikZcpAGPrf4Rmx+7BV1b8R8Ooioi4qv2WCOTgLhU9eohCjLNPmSNOi2GdG+Fiav3YuPe07hYWo5V427GiYJSrHrkZpTZK1BsdcCo1+JUoQUAEGXSs2dL5M3To2UPlqhOmn3IRpr06N8pAde1jsHBc0UY0CUR17eNQY+2MbhoKcfA13de8z27nxnMkKXQUZvHc67u0RJRrTT7e7IXLXa8/NlBHLlQggdvbo9vc/KRtfsYHE6X2qURNQ18PIdIMc2+J+sZLh6V3h7Xt43BYJGEuAgjwo06FFvd29m1bRGGv9zSAd1aRePwhRLotBqVqyYKAE8P1vvxHIALThA1omYfspEmPW7tkgi9TgOTXlflNQB0TIzEm/elYcGXv+Gtfx5G75RY3NmjlcpVEwUAH88hUlyzHy4G3EHrCVXv1wDw+G1dMOejA/j6SD7KyivwbU4BJq7ei4ISm1rlEgUGH88hUlxIhKwvUSY9bkqNw69ni6qcN58rhrWcyyxSMxcR734cJzLJfex5PIe9WKJGE9IhGxdpRJhBh4SrZhLHRxph0FW9L3ux1I5ThRZcLLUHskQi5fHxHCLFhHTIAu6gXXj/jQgzuJvCpNdi4f1p1zzCU2Jz4Pev7UCJzaFGmUTK8fRo2YMlanTNfuKTPwadFjd3jMeOpwbhsrUc0SYDNHAh97IVABBu0KGsvAL5lfdoPV+5YAUREfkT8iELAGEGHRxOF4Yu2o2rH5/dOWMQBr2x88rx8CXfAuCCFURE5B9D1kt161OEG3XY/cxg5JfYMHzJt9jy2C1IjDIhysSmIyKimjEp/LA7nGgfH3HlODHKVOWYiIjIl4CHrBBiOID/K6X8U+VxfwCLADgAfCmlfDHQNdVGlEmP3c8MZg+WiIhqLaCzi4UQiwDMv+rnvgfgTwAGAOgnhOgTyJpqKy7SiPbxEbwPS0R1dsl2CWdKzuCS7ZLapVCABbpb9i2AjwA8CgBCiBgAJillTuXxFwCGAPjR1wVsNhvMZnOjF9YiqS3+Z/oAAEBEmBFl5RUIN+jgdNgb9edZrVZF6m8u2D6+sW1q1pTbJ7xVOIZ9MgyfDPsE546eC/jPt1qtAf+Z5KZIyOswJ2sAAAb/SURBVAohxgGYftXpsVLK9UKIQV7nYgB4L7dUDKBTTdc2mUzo3r17o9Tpy6lCCwa/sQu7nxmM9gkt0DahRaNd22w2K15/MGP7+Ma2qVlTbJ9LtksoLS9FQVkBAKCooggJ7RMQaYhErCk2YHWYzWZYLJaA/Tz6D0VCVkq5HMDyWny0CID3MjPRAFQbT7lYakeJzcFnYomoUZSWlyJjU8aV49HbRgMAskdmBzRkST2qzuKRUhYJIexCiM4AjgK4E4BqE588qzp58JlYImqISEMkskdmo6CsAKO3jcaHQz9EQri7J0uhoSlMlZ0I4EMAOrhnF3+vViGeGcR8JpaIGkOsKbZKjzUhPAHJUckqVkSBFvD0kFLuBLDT6/g7AP0DXUd14iKNVXqsfCaWiBqDp0fLHmzoYRetGnwmloga09U9WgodTJFqXN2jJSIiqo+Q3+qOiIhIKQxZIiIihTBkiYiIFMKQJSIiUghDloiISCEMWSIiIoUwZImIiBTCkCUiIlIIQ5aIiEghDFkiIiKFMGSJiIgUwpAlIiJSCEOWiIhIIQxZIiIihTBkiYiIFMKQJSIiUghDloiISCF6tQsgIiL/LtkuobS8FJGGSMSaYtUuh2qJPVkioiBQWl6KjE0ZKC0vVbsUqgP2ZImImjBPD7agrAAArnxljzY4MGSJiJowTw/WY/S20QCA7JHZDNkgwJAlImrCIg2RyB6ZjYKyAozeNhofDv0QCeEJiDREql0a1QJDloioCYs1xVbpsSaEJyA5KlnFiqguOPGJiCgIeHq07MEGF/ZkiYiCwNU9WgoO7MkSEREphCFLRESkEIYsERGRQhiyRERECmHIEhERKUTjcrnUrqFO9u7dmwfghNp1EBEFmdT09PSWahcRaoIuZImIiIIFh4uJiIgUwpAlIiJSCEOWiIhIIQxZIiIihTBkiYiIFMKQJSIiUgh34QkAIUQLAKsBxAAwAnhSSvkvIUR/AIsAOAB8KaV8UcUyVSOE0AJYAiANgA3AI1LKI+pWpS4hhAHACgAdAJgAzAVwEMAHAFwADgCYLKV0qlSi6oQQSQD2AvgD3P8f+gBsGwCAEGIWgD/C/e/NEgC7wPZRBXuygfEkgO1SyoEAxgD4e+X59wD8CcAAAP2EEH3UKU919wIIk1L+DsBMAAtUrqcpeAhAgZTy9wDuAvAOgDcBzKk8pwFwj4r1qaryl5D/B6Cs8hTbppIQYhCAWwDcCmAggPZg+6iGIRsYC+H+BwFwjx5YhRAxAExSyhwppQvAFwCGqFWgygYAyAYAKeV3APqqW06T8N8AMr2OHQDS4e6RAMDnAG4PdFFNyBtw/5J6tvKYbfMfdwL4BcAWAJ8A+BRsH9VwuLiRCSHGAZh+1emxUso9QojWcA8bT4N76LjI6zPFADoFpsomJwbAZa/jCiGEXkrpUKsgtUkpSwBACBENYCOAOQDeqPyFDHD/fWmhUnmqEkKMAZAnpfyiclgUADRsmysSAaQCuBtARwAfA9CyfdTBkG1kUsrlAJZffV4I0RPAOgAzpJS7Knuy0V4fiQZwKTBVNjlFqNoW2lAOWA8hRHu4eyNLpJRrhBCveb0dyn9f/grAJYS4HcCNAFYCSPJ6P5TbBgAKABySUtoBSCGEFe4hY49Qb5+A4nBxAAghrod7+O9PUsrPAUBKWQTALoToLITQwD3Es1vFMtX0DYChAFA5GewXdctRnxCiFYAvATwrpVxRefqnyvttgPs+bUj+fZFS/peUcqCUchCAfQAeBvA52+aKrwFkCCE0Qoi2ACIBbGf7qIM92cCYDyAMwCIhBABcllLeA2AigA8B6OCeXfy9eiWqaguAPwghvoV7UsZYletpCmYDiAOQKYTw3JudCuBtIYQRgBnuYWRyewrAMrYNIKX8VAjxXwB+gLsjNRnAMbB9VMFdeIiIiBTC4WIiIiKFMGSJiIgUwpAlIiJSCEOWiIhIIQxZIiIihfARHqJaqnzOcAPcC/W7AIQD+FBKuVgIMQHu9YadAAwAnpNS7vT63mkAWkspZwa6biJSD0OWqG7+R0r5AAAIIUxwr6iTB/dOMEOklOVCiI4A/lcI0RtAKYBlAPoB2KRW0USkDg4XE9VfNIAKAI8CmCelLAcAKeUxADdKKfPhXoRkJYCXVauSiFTDnixR3dwmhNgJ97BwOYApcO+ydNT7Q1LKgsqvFwF8WbmoPRGFGIYsUd1cGS72qLzf2h5eOwkJIe4A8LOUMjfA9RFRE8LhYqKGWwH3GsN6ABBCdIN7JyanqlURkerYkyVqICnlOiFEGwBfCyHscG/48JCU8oLKpRGRyrhBABERkUI4XExERKQQhiwREZFCGLJEREQKYcgSEREphCFLRESkEIYsERGRQhiyRERECvn/Jhym6RMCF9QAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "sns.scatterplot(vst_pca['PC1'],vst_pca['PC2'], hue_order = ['4H', '12H', '24H'], hue=metadata['Time'],\n",
    "                style=metadata['Treatment'], style_order=['untr','mock','S1','S2'])\n",
    "plt.legend(bbox_to_anchor=(1, 1))\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "dnames=['gse28166','gse33267','gse37571','gse45042','gse56677']+['gse147507']*11+['gse148729']*4\n",
    "fnames=['gse28166_DE','gse33267_DE','gse37571_DE','gse45042_DE','gse56677_DE']+\\\n",
    "        ['series1_DE','series2_DE','series3_DE','series4_DE','series5_DE','series6_DE','series7_DE','series8a_DE']+\\\n",
    "        ['series8b_DE','series9_DE','series15_DE']+\\\n",
    "        ['gse148729_DE_mock_S1_4H', 'gse148729_DE_mock_S2_4H', 'gse148729_DE_mock_S1_24H', 'gse148729_DE_mock_S2_24H']\n",
    "data={}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] File ../data/gse28166/gse28166_DE.csv does not exist: '../data/gse28166/gse28166_DE.csv'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-83-c440d675c55a>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdnames\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m     \u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mfnames\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'../data/%s/%s.csv'\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mdnames\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mfnames\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0msep\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m','\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mheader\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mindex_col\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36mparser_f\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, dialect, error_bad_lines, warn_bad_lines, delim_whitespace, low_memory, memory_map, float_precision)\u001b[0m\n\u001b[1;32m    674\u001b[0m         )\n\u001b[1;32m    675\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 676\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0m_read\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    677\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    678\u001b[0m     \u001b[0mparser_f\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__name__\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    446\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    447\u001b[0m     \u001b[0;31m# Create the parser.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 448\u001b[0;31m     \u001b[0mparser\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mTextFileReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfp_or_buf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    449\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    450\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mchunksize\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0miterator\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m    878\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"has_index_names\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"has_index_names\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    879\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 880\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mengine\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    881\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    882\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_make_engine\u001b[0;34m(self, engine)\u001b[0m\n\u001b[1;32m   1112\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mengine\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"c\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1113\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"c\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1114\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mCParserWrapper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1115\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1116\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"python\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, src, **kwds)\u001b[0m\n\u001b[1;32m   1889\u001b[0m         \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"usecols\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0musecols\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1890\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1891\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mparsers\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mTextReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msrc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1892\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munnamed_cols\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munnamed_cols\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1893\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader.__cinit__\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader._setup_parser_source\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] File ../data/gse28166/gse28166_DE.csv does not exist: '../data/gse28166/gse28166_DE.csv'"
     ]
    }
   ],
   "source": [
    "for i in range(len(dnames)):\n",
    "    data[fnames[i]]=pd.read_csv('../data/%s/%s.csv' % (dnames[i],fnames[i]),sep=',',header=0,index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>lfcSE</th>\n",
       "      <th>stat</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TSPAN6</th>\n",
       "      <td>835.490187</td>\n",
       "      <td>-0.227968</td>\n",
       "      <td>0.161151</td>\n",
       "      <td>-1.414620</td>\n",
       "      <td>0.157180</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DPM1</th>\n",
       "      <td>700.023036</td>\n",
       "      <td>-0.081032</td>\n",
       "      <td>0.160045</td>\n",
       "      <td>-0.506309</td>\n",
       "      <td>0.612640</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SCYL3</th>\n",
       "      <td>197.467816</td>\n",
       "      <td>0.109047</td>\n",
       "      <td>0.249498</td>\n",
       "      <td>0.437064</td>\n",
       "      <td>0.662065</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C1orf112</th>\n",
       "      <td>336.503181</td>\n",
       "      <td>0.071293</td>\n",
       "      <td>0.206622</td>\n",
       "      <td>0.345040</td>\n",
       "      <td>0.730065</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FGR</th>\n",
       "      <td>0.593175</td>\n",
       "      <td>0.394332</td>\n",
       "      <td>4.206908</td>\n",
       "      <td>0.093735</td>\n",
       "      <td>0.925320</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASDURF</th>\n",
       "      <td>88.908432</td>\n",
       "      <td>-0.062133</td>\n",
       "      <td>0.347622</td>\n",
       "      <td>-0.178737</td>\n",
       "      <td>0.858144</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RCAN3AS</th>\n",
       "      <td>5.295571</td>\n",
       "      <td>-0.170827</td>\n",
       "      <td>1.317524</td>\n",
       "      <td>-0.129658</td>\n",
       "      <td>0.896837</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DERPC</th>\n",
       "      <td>467.583834</td>\n",
       "      <td>-0.059529</td>\n",
       "      <td>0.184201</td>\n",
       "      <td>-0.323173</td>\n",
       "      <td>0.746564</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RNVU1-8</th>\n",
       "      <td>0.223785</td>\n",
       "      <td>-1.237515</td>\n",
       "      <td>4.994031</td>\n",
       "      <td>-0.247799</td>\n",
       "      <td>0.804290</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NOTCH2NLC</th>\n",
       "      <td>12.290333</td>\n",
       "      <td>0.785789</td>\n",
       "      <td>0.891517</td>\n",
       "      <td>0.881407</td>\n",
       "      <td>0.378098</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23971 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             baseMean  log2FoldChange     lfcSE      stat    pvalue  padj\n",
       "TSPAN6     835.490187       -0.227968  0.161151 -1.414620  0.157180   NaN\n",
       "DPM1       700.023036       -0.081032  0.160045 -0.506309  0.612640   NaN\n",
       "SCYL3      197.467816        0.109047  0.249498  0.437064  0.662065   NaN\n",
       "C1orf112   336.503181        0.071293  0.206622  0.345040  0.730065   NaN\n",
       "FGR          0.593175        0.394332  4.206908  0.093735  0.925320   NaN\n",
       "...               ...             ...       ...       ...       ...   ...\n",
       "ASDURF      88.908432       -0.062133  0.347622 -0.178737  0.858144   NaN\n",
       "RCAN3AS      5.295571       -0.170827  1.317524 -0.129658  0.896837   NaN\n",
       "DERPC      467.583834       -0.059529  0.184201 -0.323173  0.746564   NaN\n",
       "RNVU1-8      0.223785       -1.237515  4.994031 -0.247799  0.804290   NaN\n",
       "NOTCH2NLC   12.290333        0.785789  0.891517  0.881407  0.378098   NaN\n",
       "\n",
       "[23971 rows x 6 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data['gse148729_DE_mock_S1_4H']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter same genes from all dataset (8710 were dropped)\n",
    "genes=set(data['gse28166_DE'].index)\n",
    "for dname in data.keys():\n",
    "    genes=genes&set(data[dname].index)\n",
    "genes=list(genes)\n",
    "for dname in data.keys():\n",
    "    data[dname]=data[dname].loc[genes]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "results=pd.DataFrame(index=genes,columns=data.keys())\n",
    "for dname in results.columns:\n",
    "    if 't' in data[dname].columns:\n",
    "        results[dname]=data[dname]['t']\n",
    "    else:\n",
    "        results[dname]=data[dname]['stat']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "columns_dict = {\n",
    "    'gse28166_DE':'GSE28166_Calu3_H5N1',\n",
    "     'gse33267_DE':'GSE33267_Calu3_SARS.CoV',\n",
    "     'gse37571_DE':'GSE37571_Calu3_Influenza',\n",
    "     'gse45042_DE':'GSE45042_Calu3_MERS',\n",
    "     'gse56677_DE':'GSE56677_Calu3_MERS',\n",
    "     'series1_DE':'GSE147507.S1_NHBE_SARS.CoV.2',\n",
    "     'series2_DE':'GSE147507.S2_A549_SARS.CoV.2',\n",
    "     'series3_DE':'GSE147507.S3_A549_RSV',\n",
    "     'series4_DE':'GSE147507.S4_A549_IAV',\n",
    "     'series5_DE':'GSE147507.S5_A549_SARS.CoV.2',\n",
    "     'series6_DE':'GSE147507.S6_A549_ACE2.SARS.CoV.2',\n",
    "     'series7_DE':'GSE147507.S7_Calu3_SARS.CoV.2',\n",
    "     'series8a_DE':'GSE147507.S8_A549_RSV',\n",
    "     'series8b_DE':'GSE147507.S8_A549_HPIV',\n",
    "     'series9_DE':'GSE147507.S9_NHBE_IAV',\n",
    "     'series15_DE':'GSE147507.S15_Infected.Lung_SARS.CoV.2',\n",
    "     'gse148729_DE_mock_S1_4H':'GSE148729_Calu3_SARS.CoV_4H',\n",
    "     'gse148729_DE_mock_S2_4H':'GSE148729_Calu3_SARS.CoV_24H',\n",
    "     'gse148729_DE_mock_S1_24H':'GSE148729_Calu3_SARS.CoV.2_4H',\n",
    "     'gse148729_DE_mock_S2_24H':'GSE148729_Calu3_SARS.CoV.2_24H'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "results = results.rename(columns=columns_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "results.to_csv('../results/microarray_and_rnaseq_expression.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GSE28166_Calu3_H5N1</th>\n",
       "      <th>GSE33267_Calu3_SARS.CoV</th>\n",
       "      <th>GSE37571_Calu3_Influenza</th>\n",
       "      <th>GSE45042_Calu3_MERS</th>\n",
       "      <th>GSE56677_Calu3_MERS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TOR1AIP1</th>\n",
       "      <td>-0.265855</td>\n",
       "      <td>-1.506547</td>\n",
       "      <td>4.691753</td>\n",
       "      <td>4.457274</td>\n",
       "      <td>-0.901465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TMEM126A</th>\n",
       "      <td>1.719826</td>\n",
       "      <td>-1.776039</td>\n",
       "      <td>-2.975716</td>\n",
       "      <td>-4.269872</td>\n",
       "      <td>-1.542372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BRD8</th>\n",
       "      <td>-12.437209</td>\n",
       "      <td>2.327074</td>\n",
       "      <td>-9.176630</td>\n",
       "      <td>1.118141</td>\n",
       "      <td>-1.332901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DAPP1</th>\n",
       "      <td>-13.756992</td>\n",
       "      <td>7.007527</td>\n",
       "      <td>4.655702</td>\n",
       "      <td>11.564376</td>\n",
       "      <td>5.867346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SLC27A2</th>\n",
       "      <td>-19.058259</td>\n",
       "      <td>-1.693200</td>\n",
       "      <td>-3.348291</td>\n",
       "      <td>-3.229593</td>\n",
       "      <td>-5.023152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TNFAIP8</th>\n",
       "      <td>-5.949700</td>\n",
       "      <td>-2.491375</td>\n",
       "      <td>3.565728</td>\n",
       "      <td>1.595638</td>\n",
       "      <td>-2.186632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FZD7</th>\n",
       "      <td>-6.987377</td>\n",
       "      <td>-1.405584</td>\n",
       "      <td>-0.093153</td>\n",
       "      <td>-9.756436</td>\n",
       "      <td>-0.542419</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPS4X</th>\n",
       "      <td>-3.091146</td>\n",
       "      <td>-1.805310</td>\n",
       "      <td>-4.770010</td>\n",
       "      <td>-0.648007</td>\n",
       "      <td>-2.558815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TMSB15B</th>\n",
       "      <td>6.332637</td>\n",
       "      <td>1.642774</td>\n",
       "      <td>-5.477011</td>\n",
       "      <td>4.329035</td>\n",
       "      <td>0.682380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>POLR3D</th>\n",
       "      <td>-4.440179</td>\n",
       "      <td>-0.695539</td>\n",
       "      <td>3.294593</td>\n",
       "      <td>9.153592</td>\n",
       "      <td>2.625260</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          GSE28166_Calu3_H5N1  GSE33267_Calu3_SARS.CoV  \\\n",
       "TOR1AIP1            -0.265855                -1.506547   \n",
       "TMEM126A             1.719826                -1.776039   \n",
       "BRD8               -12.437209                 2.327074   \n",
       "DAPP1              -13.756992                 7.007527   \n",
       "SLC27A2            -19.058259                -1.693200   \n",
       "TNFAIP8             -5.949700                -2.491375   \n",
       "FZD7                -6.987377                -1.405584   \n",
       "RPS4X               -3.091146                -1.805310   \n",
       "TMSB15B              6.332637                 1.642774   \n",
       "POLR3D              -4.440179                -0.695539   \n",
       "\n",
       "          GSE37571_Calu3_Influenza  GSE45042_Calu3_MERS  GSE56677_Calu3_MERS  \n",
       "TOR1AIP1                  4.691753             4.457274            -0.901465  \n",
       "TMEM126A                 -2.975716            -4.269872            -1.542372  \n",
       "BRD8                     -9.176630             1.118141            -1.332901  \n",
       "DAPP1                     4.655702            11.564376             5.867346  \n",
       "SLC27A2                  -3.348291            -3.229593            -5.023152  \n",
       "TNFAIP8                   3.565728             1.595638            -2.186632  \n",
       "FZD7                     -0.093153            -9.756436            -0.542419  \n",
       "RPS4X                    -4.770010            -0.648007            -2.558815  \n",
       "TMSB15B                  -5.477011             4.329035             0.682380  \n",
       "POLR3D                    3.294593             9.153592             2.625260  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results.iloc[:10, :5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
